// react-search-box/dist/react-search-box.es.js
import Pe, { useRef as st, useContext as it, useLayoutEffect as Jr, useDebugValue as Vt, createElement as Qr, useEffect as lr, useState as ft } from "react";
var et = {}, Zr = {
  get exports() {
    return et;
  },
  set exports(e) {
    et = e;
  }
}, Be = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Wt;
function qr() {
  if (Wt)
    return Be;
  Wt = 1;
  var e = Pe, t = Symbol.for("react.element"), r = Symbol.for("react.fragment"), n = Object.prototype.hasOwnProperty, o = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, i = { key: !0, ref: !0, __self: !0, __source: !0 };
  function a(c, l, f) {
    var d, h = {}, y = null, w = null;
    f !== void 0 && (y = "" + f), l.key !== void 0 && (y = "" + l.key), l.ref !== void 0 && (w = l.ref);
    for (d in l)
      n.call(l, d) && !i.hasOwnProperty(d) && (h[d] = l[d]);
    if (c && c.defaultProps)
      for (d in l = c.defaultProps, l)
        h[d] === void 0 && (h[d] = l[d]);
    return { $$typeof: t, type: c, key: y, ref: w, props: h, _owner: o.current };
  }
  return Be.Fragment = r, Be.jsx = a, Be.jsxs = a, Be;
}
var Ue = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Bt;
function en() {
  return Bt || (Bt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = Pe, t = Symbol.for("react.element"), r = Symbol.for("react.portal"), n = Symbol.for("react.fragment"), o = Symbol.for("react.strict_mode"), i = Symbol.for("react.profiler"), a = Symbol.for("react.provider"), c = Symbol.for("react.context"), l = Symbol.for("react.forward_ref"), f = Symbol.for("react.suspense"), d = Symbol.for("react.suspense_list"), h = Symbol.for("react.memo"), y = Symbol.for("react.lazy"), w = Symbol.for("react.offscreen"), b = Symbol.iterator, A = "@@iterator";
    function R(s) {
      if (s === null || typeof s != "object")
        return null;
      var m = b && s[b] || s[A];
      return typeof m == "function" ? m : null;
    }
    var O = e.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function _(s) {
      {
        for (var m = arguments.length, S = new Array(m > 1 ? m - 1 : 0), I = 1; I < m; I++)
          S[I - 1] = arguments[I];
        T("error", s, S);
      }
    }
    function T(s, m, S) {
      {
        var I = O.ReactDebugCurrentFrame, W = I.getStackAddendum();
        W !== "" && (m += "%s", S = S.concat([W]));
        var ee = S.map(function(z) {
          return String(z);
        });
        ee.unshift("Warning: " + m), Function.prototype.apply.call(console[s], console, ee);
      }
    }
    var k = !1, g = !1, X = !1, J = !1, G = !1, Q;
    Q = Symbol.for("react.module.reference");
    function fe(s) {
      return !!(typeof s == "string" || typeof s == "function" || s === n || s === i || G || s === o || s === f || s === d || J || s === w || k || g || X || typeof s == "object" && s !== null && (s.$$typeof === y || s.$$typeof === h || s.$$typeof === a || s.$$typeof === c || s.$$typeof === l || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      s.$$typeof === Q || s.getModuleId !== void 0));
    }
    function de(s, m, S) {
      var I = s.displayName;
      if (I)
        return I;
      var W = m.displayName || m.name || "";
      return W !== "" ? S + "(" + W + ")" : S;
    }
    function Z(s) {
      return s.displayName || "Context";
    }
    function q(s) {
      if (s == null)
        return null;
      if (typeof s.tag == "number" && _("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof s == "function")
        return s.displayName || s.name || null;
      if (typeof s == "string")
        return s;
      switch (s) {
        case n:
          return "Fragment";
        case r:
          return "Portal";
        case i:
          return "Profiler";
        case o:
          return "StrictMode";
        case f:
          return "Suspense";
        case d:
          return "SuspenseList";
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case c:
            var m = s;
            return Z(m) + ".Consumer";
          case a:
            var S = s;
            return Z(S._context) + ".Provider";
          case l:
            return de(s, s.render, "ForwardRef");
          case h:
            var I = s.displayName || null;
            return I !== null ? I : q(s.type) || "Memo";
          case y: {
            var W = s, ee = W._payload, z = W._init;
            try {
              return q(z(ee));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var Y = Object.assign, H = 0, he, ne, te, ve, ye, Se, C;
    function E() {
    }
    E.__reactDisabledLog = !0;
    function x() {
      {
        if (H === 0) {
          he = console.log, ne = console.info, te = console.warn, ve = console.error, ye = console.group, Se = console.groupCollapsed, C = console.groupEnd;
          var s = {
            configurable: !0,
            enumerable: !0,
            value: E,
            writable: !0
          };
          Object.defineProperties(console, {
            info: s,
            log: s,
            warn: s,
            error: s,
            group: s,
            groupCollapsed: s,
            groupEnd: s
          });
        }
        H++;
      }
    }
    function M() {
      {
        if (H--, H === 0) {
          var s = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: Y({}, s, {
              value: he
            }),
            info: Y({}, s, {
              value: ne
            }),
            warn: Y({}, s, {
              value: te
            }),
            error: Y({}, s, {
              value: ve
            }),
            group: Y({}, s, {
              value: ye
            }),
            groupCollapsed: Y({}, s, {
              value: Se
            }),
            groupEnd: Y({}, s, {
              value: C
            })
          });
        }
        H < 0 && _("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var u = O.ReactCurrentDispatcher, D;
    function v(s, m, S) {
      {
        if (D === void 0)
          try {
            throw Error();
          } catch (W) {
            var I = W.stack.trim().match(/\n( *(at )?)/);
            D = I && I[1] || "";
          }
        return `
` + D + s;
      }
    }
    var K = !1, j;
    {
      var F = typeof WeakMap == "function" ? WeakMap : Map;
      j = new F();
    }
    function p(s, m) {
      if (!s || K)
        return "";
      {
        var S = j.get(s);
        if (S !== void 0)
          return S;
      }
      var I;
      K = !0;
      var W = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var ee;
      ee = u.current, u.current = null, x();
      try {
        if (m) {
          var z = function() {
            throw Error();
          };
          if (Object.defineProperty(z.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(z, []);
            } catch (Ae) {
              I = Ae;
            }
            Reflect.construct(s, [], z);
          } else {
            try {
              z.call();
            } catch (Ae) {
              I = Ae;
            }
            s.call(z.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (Ae) {
            I = Ae;
          }
          s();
        }
      } catch (Ae) {
        if (Ae && I && typeof Ae.stack == "string") {
          for (var L = Ae.stack.split(`
`), pe = I.stack.split(`
`), ae = L.length - 1, le = pe.length - 1; ae >= 1 && le >= 0 && L[ae] !== pe[le]; )
            le--;
          for (; ae >= 1 && le >= 0; ae--, le--)
            if (L[ae] !== pe[le]) {
              if (ae !== 1 || le !== 1)
                do
                  if (ae--, le--, le < 0 || L[ae] !== pe[le]) {
                    var be = `
` + L[ae].replace(" at new ", " at ");
                    return s.displayName && be.includes("<anonymous>") && (be = be.replace("<anonymous>", s.displayName)), typeof s == "function" && j.set(s, be), be;
                  }
                while (ae >= 1 && le >= 0);
              break;
            }
        }
      } finally {
        K = !1, u.current = ee, M(), Error.prepareStackTrace = W;
      }
      var Fe = s ? s.displayName || s.name : "", zt = Fe ? v(Fe) : "";
      return typeof s == "function" && j.set(s, zt), zt;
    }
    function oe(s, m, S) {
      return p(s, !1);
    }
    function se(s) {
      var m = s.prototype;
      return !!(m && m.isReactComponent);
    }
    function $(s, m, S) {
      if (s == null)
        return "";
      if (typeof s == "function")
        return p(s, se(s));
      if (typeof s == "string")
        return v(s);
      switch (s) {
        case f:
          return v("Suspense");
        case d:
          return v("SuspenseList");
      }
      if (typeof s == "object")
        switch (s.$$typeof) {
          case l:
            return oe(s.render);
          case h:
            return $(s.type, m, S);
          case y: {
            var I = s, W = I._payload, ee = I._init;
            try {
              return $(ee(W), m, S);
            } catch {
            }
          }
        }
      return "";
    }
    var ie = Object.prototype.hasOwnProperty, V = {}, ce = O.ReactDebugCurrentFrame;
    function Ee(s) {
      if (s) {
        var m = s._owner, S = $(s.type, s._source, m ? m.type : null);
        ce.setExtraStackFrame(S);
      } else
        ce.setExtraStackFrame(null);
    }
    function ue(s, m, S, I, W) {
      {
        var ee = Function.call.bind(ie);
        for (var z in s)
          if (ee(s, z)) {
            var L = void 0;
            try {
              if (typeof s[z] != "function") {
                var pe = Error((I || "React class") + ": " + S + " type `" + z + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof s[z] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw pe.name = "Invariant Violation", pe;
              }
              L = s[z](m, z, I, S, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (ae) {
              L = ae;
            }
            L && !(L instanceof Error) && (Ee(W), _("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", I || "React class", S, z, typeof L), Ee(null)), L instanceof Error && !(L.message in V) && (V[L.message] = !0, Ee(W), _("Failed %s type: %s", S, L.message), Ee(null));
          }
      }
    }
    var je = Array.isArray;
    function Ce(s) {
      return je(s);
    }
    function me(s) {
      {
        var m = typeof Symbol == "function" && Symbol.toStringTag, S = m && s[Symbol.toStringTag] || s.constructor.name || "Object";
        return S;
      }
    }
    function P(s) {
      try {
        return re(s), !1;
      } catch {
        return !0;
      }
    }
    function re(s) {
      return "" + s;
    }
    function De(s) {
      if (P(s))
        return _("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", me(s)), re(s);
    }
    var Re = O.ReactCurrentOwner, we = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Mt, Nt, at;
    at = {};
    function $r(s) {
      if (ie.call(s, "ref")) {
        var m = Object.getOwnPropertyDescriptor(s, "ref").get;
        if (m && m.isReactWarning)
          return !1;
      }
      return s.ref !== void 0;
    }
    function jr(s) {
      if (ie.call(s, "key")) {
        var m = Object.getOwnPropertyDescriptor(s, "key").get;
        if (m && m.isReactWarning)
          return !1;
      }
      return s.key !== void 0;
    }
    function Dr(s, m) {
      if (typeof s.ref == "string" && Re.current && m && Re.current.stateNode !== m) {
        var S = q(Re.current.type);
        at[S] || (_('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', q(Re.current.type), s.ref), at[S] = !0);
      }
    }
    function Lr(s, m) {
      {
        var S = function() {
          Mt || (Mt = !0, _("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", m));
        };
        S.isReactWarning = !0, Object.defineProperty(s, "key", {
          get: S,
          configurable: !0
        });
      }
    }
    function Fr(s, m) {
      {
        var S = function() {
          Nt || (Nt = !0, _("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", m));
        };
        S.isReactWarning = !0, Object.defineProperty(s, "ref", {
          get: S,
          configurable: !0
        });
      }
    }
    var Yr = function(s, m, S, I, W, ee, z) {
      var L = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: t,
        // Built-in properties that belong on the element
        type: s,
        key: m,
        ref: S,
        props: z,
        // Record the component responsible for creating this element.
        _owner: ee
      };
      return L._store = {}, Object.defineProperty(L._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(L, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: I
      }), Object.defineProperty(L, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: W
      }), Object.freeze && (Object.freeze(L.props), Object.freeze(L)), L;
    };
    function zr(s, m, S, I, W) {
      {
        var ee, z = {}, L = null, pe = null;
        S !== void 0 && (De(S), L = "" + S), jr(m) && (De(m.key), L = "" + m.key), $r(m) && (pe = m.ref, Dr(m, W));
        for (ee in m)
          ie.call(m, ee) && !we.hasOwnProperty(ee) && (z[ee] = m[ee]);
        if (s && s.defaultProps) {
          var ae = s.defaultProps;
          for (ee in ae)
            z[ee] === void 0 && (z[ee] = ae[ee]);
        }
        if (L || pe) {
          var le = typeof s == "function" ? s.displayName || s.name || "Unknown" : s;
          L && Lr(z, le), pe && Fr(z, le);
        }
        return Yr(s, L, pe, W, I, Re.current, z);
      }
    }
    var ct = O.ReactCurrentOwner, $t = O.ReactDebugCurrentFrame;
    function Le(s) {
      if (s) {
        var m = s._owner, S = $(s.type, s._source, m ? m.type : null);
        $t.setExtraStackFrame(S);
      } else
        $t.setExtraStackFrame(null);
    }
    var lt;
    lt = !1;
    function ut(s) {
      return typeof s == "object" && s !== null && s.$$typeof === t;
    }
    function jt() {
      {
        if (ct.current) {
          var s = q(ct.current.type);
          if (s)
            return `

Check the render method of \`` + s + "`.";
        }
        return "";
      }
    }
    function Vr(s) {
      {
        if (s !== void 0) {
          var m = s.fileName.replace(/^.*[\\\/]/, ""), S = s.lineNumber;
          return `

Check your code at ` + m + ":" + S + ".";
        }
        return "";
      }
    }
    var Dt = {};
    function Wr(s) {
      {
        var m = jt();
        if (!m) {
          var S = typeof s == "string" ? s : s.displayName || s.name;
          S && (m = `

Check the top-level render call using <` + S + ">.");
        }
        return m;
      }
    }
    function Lt(s, m) {
      {
        if (!s._store || s._store.validated || s.key != null)
          return;
        s._store.validated = !0;
        var S = Wr(m);
        if (Dt[S])
          return;
        Dt[S] = !0;
        var I = "";
        s && s._owner && s._owner !== ct.current && (I = " It was passed a child from " + q(s._owner.type) + "."), Le(s), _('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', S, I), Le(null);
      }
    }
    function Ft(s, m) {
      {
        if (typeof s != "object")
          return;
        if (Ce(s))
          for (var S = 0; S < s.length; S++) {
            var I = s[S];
            ut(I) && Lt(I, m);
          }
        else if (ut(s))
          s._store && (s._store.validated = !0);
        else if (s) {
          var W = R(s);
          if (typeof W == "function" && W !== s.entries)
            for (var ee = W.call(s), z; !(z = ee.next()).done; )
              ut(z.value) && Lt(z.value, m);
        }
      }
    }
    function Br(s) {
      {
        var m = s.type;
        if (m == null || typeof m == "string")
          return;
        var S;
        if (typeof m == "function")
          S = m.propTypes;
        else if (typeof m == "object" && (m.$$typeof === l || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        m.$$typeof === h))
          S = m.propTypes;
        else
          return;
        if (S) {
          var I = q(m);
          ue(S, s.props, "prop", I, s);
        } else if (m.PropTypes !== void 0 && !lt) {
          lt = !0;
          var W = q(m);
          _("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", W || "Unknown");
        }
        typeof m.getDefaultProps == "function" && !m.getDefaultProps.isReactClassApproved && _("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function Ur(s) {
      {
        for (var m = Object.keys(s.props), S = 0; S < m.length; S++) {
          var I = m[S];
          if (I !== "children" && I !== "key") {
            Le(s), _("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", I), Le(null);
            break;
          }
        }
        s.ref !== null && (Le(s), _("Invalid attribute `ref` supplied to `React.Fragment`."), Le(null));
      }
    }
    function Yt(s, m, S, I, W, ee) {
      {
        var z = fe(s);
        if (!z) {
          var L = "";
          (s === void 0 || typeof s == "object" && s !== null && Object.keys(s).length === 0) && (L += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var pe = Vr(W);
          pe ? L += pe : L += jt();
          var ae;
          s === null ? ae = "null" : Ce(s) ? ae = "array" : s !== void 0 && s.$$typeof === t ? (ae = "<" + (q(s.type) || "Unknown") + " />", L = " Did you accidentally export a JSX literal instead of a component?") : ae = typeof s, _("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", ae, L);
        }
        var le = zr(s, m, S, W, ee);
        if (le == null)
          return le;
        if (z) {
          var be = m.children;
          if (be !== void 0)
            if (I)
              if (Ce(be)) {
                for (var Fe = 0; Fe < be.length; Fe++)
                  Ft(be[Fe], s);
                Object.freeze && Object.freeze(be);
              } else
                _("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ft(be, s);
        }
        return s === n ? Ur(le) : Br(le), le;
      }
    }
    function Gr(s, m, S) {
      return Yt(s, m, S, !0);
    }
    function Hr(s, m, S) {
      return Yt(s, m, S, !1);
    }
    var Kr = Hr, Xr = Gr;
    Ue.Fragment = n, Ue.jsx = Kr, Ue.jsxs = Xr;
  }()), Ue;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = qr() : e.exports = en();
})(Zr);
const ke = et.jsx, ur = et.jsxs;
function Te(e) {
  return Array.isArray ? Array.isArray(e) : hr(e) === "[object Array]";
}
const tn = 1 / 0;
function rn(e) {
  if (typeof e == "string")
    return e;
  let t = e + "";
  return t == "0" && 1 / e == -tn ? "-0" : t;
}
function nn(e) {
  return e == null ? "" : rn(e);
}
function _e(e) {
  return typeof e == "string";
}
function fr(e) {
  return typeof e == "number";
}
function on(e) {
  return e === !0 || e === !1 || sn(e) && hr(e) == "[object Boolean]";
}
function dr(e) {
  return typeof e == "object";
}
function sn(e) {
  return dr(e) && e !== null;
}
function ge(e) {
  return e != null;
}
function dt(e) {
  return !e.trim().length;
}
function hr(e) {
  return e == null ? e === void 0 ? "[object Undefined]" : "[object Null]" : Object.prototype.toString.call(e);
}
const an = "Incorrect 'index' type", cn = (e) => `Invalid value for key ${e}`, ln = (e) => `Pattern length exceeds max of ${e}.`, un = (e) => `Missing ${e} property in key`, fn = (e) => `Property 'weight' in key '${e}' must be a positive integer`, Ut = Object.prototype.hasOwnProperty;
class dn {
  constructor(t) {
    this._keys = [], this._keyMap = {};
    let r = 0;
    t.forEach((n) => {
      let o = pr(n);
      r += o.weight, this._keys.push(o), this._keyMap[o.id] = o, r += o.weight;
    }), this._keys.forEach((n) => {
      n.weight /= r;
    });
  }
  get(t) {
    return this._keyMap[t];
  }
  keys() {
    return this._keys;
  }
  toJSON() {
    return JSON.stringify(this._keys);
  }
}
function pr(e) {
  let t = null, r = null, n = null, o = 1, i = null;
  if (_e(e) || Te(e))
    n = e, t = Gt(e), r = mt(e);
  else {
    if (!Ut.call(e, "name"))
      throw new Error(un("name"));
    const a = e.name;
    if (n = a, Ut.call(e, "weight") && (o = e.weight, o <= 0))
      throw new Error(fn(a));
    t = Gt(a), r = mt(a), i = e.getFn;
  }
  return { path: t, id: r, weight: o, src: n, getFn: i };
}
function Gt(e) {
  return Te(e) ? e : e.split(".");
}
function mt(e) {
  return Te(e) ? e.join(".") : e;
}
function hn(e, t) {
  let r = [], n = !1;
  const o = (i, a, c) => {
    if (ge(i))
      if (!a[c])
        r.push(i);
      else {
        let l = a[c];
        const f = i[l];
        if (!ge(f))
          return;
        if (c === a.length - 1 && (_e(f) || fr(f) || on(f)))
          r.push(nn(f));
        else if (Te(f)) {
          n = !0;
          for (let d = 0, h = f.length; d < h; d += 1)
            o(f[d], a, c + 1);
        } else
          a.length && o(f, a, c + 1);
      }
  };
  return o(e, _e(t) ? t.split(".") : t, 0), n ? r : r[0];
}
const pn = {
  // Whether the matches should be included in the result set. When `true`, each record in the result
  // set will include the indices of the matched characters.
  // These can consequently be used for highlighting purposes.
  includeMatches: !1,
  // When `true`, the matching function will continue to the end of a search pattern even if
  // a perfect match has already been located in the string.
  findAllMatches: !1,
  // Minimum number of characters that must be matched before a result is considered a match
  minMatchCharLength: 1
}, mn = {
  // When `true`, the algorithm continues searching to the end of the input even if a perfect
  // match is found before the end of the same input.
  isCaseSensitive: !1,
  // When true, the matching function will continue to the end of a search pattern even if
  includeScore: !1,
  // List of properties that will be searched. This also supports nested properties.
  keys: [],
  // Whether to sort the result list, by score
  shouldSort: !0,
  // Default sort function: sort by ascending score, ascending index
  sortFn: (e, t) => e.score === t.score ? e.idx < t.idx ? -1 : 1 : e.score < t.score ? -1 : 1
}, gn = {
  // Approximately where in the text is the pattern expected to be found?
  location: 0,
  // At what point does the match algorithm give up. A threshold of '0.0' requires a perfect match
  // (of both letters and location), a threshold of '1.0' would match anything.
  threshold: 0.6,
  // Determines how close the match must be to the fuzzy location (specified above).
  // An exact letter match which is 'distance' characters away from the fuzzy location
  // would score as a complete mismatch. A distance of '0' requires the match be at
  // the exact location specified, a threshold of '1000' would require a perfect match
  // to be within 800 characters of the fuzzy location to be found using a 0.8 threshold.
  distance: 100
}, vn = {
  // When `true`, it enables the use of unix-like search commands
  useExtendedSearch: !1,
  // The get function to use when fetching an object's properties.
  // The default will search nested paths *ie foo.bar.baz*
  getFn: hn,
  // When `true`, search will ignore `location` and `distance`, so it won't matter
  // where in the string the pattern appears.
  // More info: https://fusejs.io/concepts/scoring-theory.html#fuzziness-score
  ignoreLocation: !1,
  // When `true`, the calculation for the relevance score (used for sorting) will
  // ignore the field-length norm.
  // More info: https://fusejs.io/concepts/scoring-theory.html#field-length-norm
  ignoreFieldNorm: !1,
  // The weight to determine how much field length norm effects scoring.
  fieldNormWeight: 1
};
var N = {
  ...mn,
  ...pn,
  ...gn,
  ...vn
};
const yn = /[^ ]+/g;
function bn(e = 1, t = 3) {
  const r = /* @__PURE__ */ new Map(), n = Math.pow(10, t);
  return {
    get(o) {
      const i = o.match(yn).length;
      if (r.has(i))
        return r.get(i);
      const a = 1 / Math.pow(i, 0.5 * e), c = parseFloat(Math.round(a * n) / n);
      return r.set(i, c), c;
    },
    clear() {
      r.clear();
    }
  };
}
class At {
  constructor({
    getFn: t = N.getFn,
    fieldNormWeight: r = N.fieldNormWeight
  } = {}) {
    this.norm = bn(r, 3), this.getFn = t, this.isCreated = !1, this.setIndexRecords();
  }
  setSources(t = []) {
    this.docs = t;
  }
  setIndexRecords(t = []) {
    this.records = t;
  }
  setKeys(t = []) {
    this.keys = t, this._keysMap = {}, t.forEach((r, n) => {
      this._keysMap[r.id] = n;
    });
  }
  create() {
    this.isCreated || !this.docs.length || (this.isCreated = !0, _e(this.docs[0]) ? this.docs.forEach((t, r) => {
      this._addString(t, r);
    }) : this.docs.forEach((t, r) => {
      this._addObject(t, r);
    }), this.norm.clear());
  }
  // Adds a doc to the end of the index
  add(t) {
    const r = this.size();
    _e(t) ? this._addString(t, r) : this._addObject(t, r);
  }
  // Removes the doc at the specified index of the index
  removeAt(t) {
    this.records.splice(t, 1);
    for (let r = t, n = this.size(); r < n; r += 1)
      this.records[r].i -= 1;
  }
  getValueForItemAtKeyId(t, r) {
    return t[this._keysMap[r]];
  }
  size() {
    return this.records.length;
  }
  _addString(t, r) {
    if (!ge(t) || dt(t))
      return;
    let n = {
      v: t,
      i: r,
      n: this.norm.get(t)
    };
    this.records.push(n);
  }
  _addObject(t, r) {
    let n = { i: r, $: {} };
    this.keys.forEach((o, i) => {
      let a = o.getFn ? o.getFn(t) : this.getFn(t, o.path);
      if (ge(a)) {
        if (Te(a)) {
          let c = [];
          const l = [{ nestedArrIndex: -1, value: a }];
          for (; l.length; ) {
            const { nestedArrIndex: f, value: d } = l.pop();
            if (ge(d))
              if (_e(d) && !dt(d)) {
                let h = {
                  v: d,
                  i: f,
                  n: this.norm.get(d)
                };
                c.push(h);
              } else
                Te(d) && d.forEach((h, y) => {
                  l.push({
                    nestedArrIndex: y,
                    value: h
                  });
                });
          }
          n.$[i] = c;
        } else if (_e(a) && !dt(a)) {
          let c = {
            v: a,
            n: this.norm.get(a)
          };
          n.$[i] = c;
        }
      }
    }), this.records.push(n);
  }
  toJSON() {
    return {
      keys: this.keys,
      records: this.records
    };
  }
}
function mr(e, t, { getFn: r = N.getFn, fieldNormWeight: n = N.fieldNormWeight } = {}) {
  const o = new At({ getFn: r, fieldNormWeight: n });
  return o.setKeys(e.map(pr)), o.setSources(t), o.create(), o;
}
function Sn(e, { getFn: t = N.getFn, fieldNormWeight: r = N.fieldNormWeight } = {}) {
  const { keys: n, records: o } = e, i = new At({ getFn: t, fieldNormWeight: r });
  return i.setKeys(n), i.setIndexRecords(o), i;
}
function Xe(e, {
  errors: t = 0,
  currentLocation: r = 0,
  expectedLocation: n = 0,
  distance: o = N.distance,
  ignoreLocation: i = N.ignoreLocation
} = {}) {
  const a = t / e.length;
  if (i)
    return a;
  const c = Math.abs(n - r);
  return o ? a + c / o : c ? 1 : a;
}
function En(e = [], t = N.minMatchCharLength) {
  let r = [], n = -1, o = -1, i = 0;
  for (let a = e.length; i < a; i += 1) {
    let c = e[i];
    c && n === -1 ? n = i : !c && n !== -1 && (o = i - 1, o - n + 1 >= t && r.push([n, o]), n = -1);
  }
  return e[i - 1] && i - n >= t && r.push([n, i - 1]), r;
}
const Me = 32;
function wn(e, t, r, {
  location: n = N.location,
  distance: o = N.distance,
  threshold: i = N.threshold,
  findAllMatches: a = N.findAllMatches,
  minMatchCharLength: c = N.minMatchCharLength,
  includeMatches: l = N.includeMatches,
  ignoreLocation: f = N.ignoreLocation
} = {}) {
  if (t.length > Me)
    throw new Error(ln(Me));
  const d = t.length, h = e.length, y = Math.max(0, Math.min(n, h));
  let w = i, b = y;
  const A = c > 1 || l, R = A ? Array(h) : [];
  let O;
  for (; (O = e.indexOf(t, b)) > -1; ) {
    let J = Xe(t, {
      currentLocation: O,
      expectedLocation: y,
      distance: o,
      ignoreLocation: f
    });
    if (w = Math.min(J, w), b = O + d, A) {
      let G = 0;
      for (; G < d; )
        R[O + G] = 1, G += 1;
    }
  }
  b = -1;
  let _ = [], T = 1, k = d + h;
  const g = 1 << d - 1;
  for (let J = 0; J < d; J += 1) {
    let G = 0, Q = k;
    for (; G < Q; )
      Xe(t, {
        errors: J,
        currentLocation: y + Q,
        expectedLocation: y,
        distance: o,
        ignoreLocation: f
      }) <= w ? G = Q : k = Q, Q = Math.floor((k - G) / 2 + G);
    k = Q;
    let fe = Math.max(1, y - Q + 1), de = a ? h : Math.min(y + Q, h) + d, Z = Array(de + 2);
    Z[de + 1] = (1 << J) - 1;
    for (let Y = de; Y >= fe; Y -= 1) {
      let H = Y - 1, he = r[e.charAt(H)];
      if (A && (R[H] = +!!he), Z[Y] = (Z[Y + 1] << 1 | 1) & he, J && (Z[Y] |= (_[Y + 1] | _[Y]) << 1 | 1 | _[Y + 1]), Z[Y] & g && (T = Xe(t, {
        errors: J,
        currentLocation: H,
        expectedLocation: y,
        distance: o,
        ignoreLocation: f
      }), T <= w)) {
        if (w = T, b = H, b <= y)
          break;
        fe = Math.max(1, 2 * y - b);
      }
    }
    if (Xe(t, {
      errors: J + 1,
      currentLocation: y,
      expectedLocation: y,
      distance: o,
      ignoreLocation: f
    }) > w)
      break;
    _ = Z;
  }
  const X = {
    isMatch: b >= 0,
    // Count exact matches (those with a score of 0) to be "almost" exact
    score: Math.max(1e-3, T)
  };
  if (A) {
    const J = En(R, c);
    J.length ? l && (X.indices = J) : X.isMatch = !1;
  }
  return X;
}
function _n(e) {
  let t = {};
  for (let r = 0, n = e.length; r < n; r += 1) {
    const o = e.charAt(r);
    t[o] = (t[o] || 0) | 1 << n - r - 1;
  }
  return t;
}
class gr {
  constructor(t, {
    location: r = N.location,
    threshold: n = N.threshold,
    distance: o = N.distance,
    includeMatches: i = N.includeMatches,
    findAllMatches: a = N.findAllMatches,
    minMatchCharLength: c = N.minMatchCharLength,
    isCaseSensitive: l = N.isCaseSensitive,
    ignoreLocation: f = N.ignoreLocation
  } = {}) {
    if (this.options = {
      location: r,
      threshold: n,
      distance: o,
      includeMatches: i,
      findAllMatches: a,
      minMatchCharLength: c,
      isCaseSensitive: l,
      ignoreLocation: f
    }, this.pattern = l ? t : t.toLowerCase(), this.chunks = [], !this.pattern.length)
      return;
    const d = (y, w) => {
      this.chunks.push({
        pattern: y,
        alphabet: _n(y),
        startIndex: w
      });
    }, h = this.pattern.length;
    if (h > Me) {
      let y = 0;
      const w = h % Me, b = h - w;
      for (; y < b; )
        d(this.pattern.substr(y, Me), y), y += Me;
      if (w) {
        const A = h - Me;
        d(this.pattern.substr(A), A);
      }
    } else
      d(this.pattern, 0);
  }
  searchIn(t) {
    const { isCaseSensitive: r, includeMatches: n } = this.options;
    if (r || (t = t.toLowerCase()), this.pattern === t) {
      let b = {
        isMatch: !0,
        score: 0
      };
      return n && (b.indices = [[0, t.length - 1]]), b;
    }
    const {
      location: o,
      distance: i,
      threshold: a,
      findAllMatches: c,
      minMatchCharLength: l,
      ignoreLocation: f
    } = this.options;
    let d = [], h = 0, y = !1;
    this.chunks.forEach(({ pattern: b, alphabet: A, startIndex: R }) => {
      const { isMatch: O, score: _, indices: T } = wn(t, b, A, {
        location: o + R,
        distance: i,
        threshold: a,
        findAllMatches: c,
        minMatchCharLength: l,
        includeMatches: n,
        ignoreLocation: f
      });
      O && (y = !0), h += _, O && T && (d = [...d, ...T]);
    });
    let w = {
      isMatch: y,
      score: y ? h / this.chunks.length : 1
    };
    return y && n && (w.indices = d), w;
  }
}
class Ie {
  constructor(t) {
    this.pattern = t;
  }
  static isMultiMatch(t) {
    return Ht(t, this.multiRegex);
  }
  static isSingleMatch(t) {
    return Ht(t, this.singleRegex);
  }
  search() {
  }
}
function Ht(e, t) {
  const r = e.match(t);
  return r ? r[1] : null;
}
class xn extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "exact";
  }
  static get multiRegex() {
    return /^="(.*)"$/;
  }
  static get singleRegex() {
    return /^=(.*)$/;
  }
  search(t) {
    const r = t === this.pattern;
    return {
      isMatch: r,
      score: r ? 0 : 1,
      indices: [0, this.pattern.length - 1]
    };
  }
}
class Cn extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "inverse-exact";
  }
  static get multiRegex() {
    return /^!"(.*)"$/;
  }
  static get singleRegex() {
    return /^!(.*)$/;
  }
  search(t) {
    const n = t.indexOf(this.pattern) === -1;
    return {
      isMatch: n,
      score: n ? 0 : 1,
      indices: [0, t.length - 1]
    };
  }
}
class Rn extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "prefix-exact";
  }
  static get multiRegex() {
    return /^\^"(.*)"$/;
  }
  static get singleRegex() {
    return /^\^(.*)$/;
  }
  search(t) {
    const r = t.startsWith(this.pattern);
    return {
      isMatch: r,
      score: r ? 0 : 1,
      indices: [0, this.pattern.length - 1]
    };
  }
}
class An extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "inverse-prefix-exact";
  }
  static get multiRegex() {
    return /^!\^"(.*)"$/;
  }
  static get singleRegex() {
    return /^!\^(.*)$/;
  }
  search(t) {
    const r = !t.startsWith(this.pattern);
    return {
      isMatch: r,
      score: r ? 0 : 1,
      indices: [0, t.length - 1]
    };
  }
}
class kn extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "suffix-exact";
  }
  static get multiRegex() {
    return /^"(.*)"\$$/;
  }
  static get singleRegex() {
    return /^(.*)\$$/;
  }
  search(t) {
    const r = t.endsWith(this.pattern);
    return {
      isMatch: r,
      score: r ? 0 : 1,
      indices: [t.length - this.pattern.length, t.length - 1]
    };
  }
}
class Tn extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "inverse-suffix-exact";
  }
  static get multiRegex() {
    return /^!"(.*)"\$$/;
  }
  static get singleRegex() {
    return /^!(.*)\$$/;
  }
  search(t) {
    const r = !t.endsWith(this.pattern);
    return {
      isMatch: r,
      score: r ? 0 : 1,
      indices: [0, t.length - 1]
    };
  }
}
class vr extends Ie {
  constructor(t, {
    location: r = N.location,
    threshold: n = N.threshold,
    distance: o = N.distance,
    includeMatches: i = N.includeMatches,
    findAllMatches: a = N.findAllMatches,
    minMatchCharLength: c = N.minMatchCharLength,
    isCaseSensitive: l = N.isCaseSensitive,
    ignoreLocation: f = N.ignoreLocation
  } = {}) {
    super(t), this._bitapSearch = new gr(t, {
      location: r,
      threshold: n,
      distance: o,
      includeMatches: i,
      findAllMatches: a,
      minMatchCharLength: c,
      isCaseSensitive: l,
      ignoreLocation: f
    });
  }
  static get type() {
    return "fuzzy";
  }
  static get multiRegex() {
    return /^"(.*)"$/;
  }
  static get singleRegex() {
    return /^(.*)$/;
  }
  search(t) {
    return this._bitapSearch.searchIn(t);
  }
}
class yr extends Ie {
  constructor(t) {
    super(t);
  }
  static get type() {
    return "include";
  }
  static get multiRegex() {
    return /^'"(.*)"$/;
  }
  static get singleRegex() {
    return /^'(.*)$/;
  }
  search(t) {
    let r = 0, n;
    const o = [], i = this.pattern.length;
    for (; (n = t.indexOf(this.pattern, r)) > -1; )
      r = n + i, o.push([n, r - 1]);
    const a = !!o.length;
    return {
      isMatch: a,
      score: a ? 0 : 1,
      indices: o
    };
  }
}
const gt = [
  xn,
  yr,
  Rn,
  An,
  Tn,
  kn,
  Cn,
  vr
], Kt = gt.length, On = / +(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)/, Pn = "|";
function In(e, t = {}) {
  return e.split(Pn).map((r) => {
    let n = r.trim().split(On).filter((i) => i && !!i.trim()), o = [];
    for (let i = 0, a = n.length; i < a; i += 1) {
      const c = n[i];
      let l = !1, f = -1;
      for (; !l && ++f < Kt; ) {
        const d = gt[f];
        let h = d.isMultiMatch(c);
        h && (o.push(new d(h, t)), l = !0);
      }
      if (!l)
        for (f = -1; ++f < Kt; ) {
          const d = gt[f];
          let h = d.isSingleMatch(c);
          if (h) {
            o.push(new d(h, t));
            break;
          }
        }
    }
    return o;
  });
}
const Mn = /* @__PURE__ */ new Set([vr.type, yr.type]);
class Nn {
  constructor(t, {
    isCaseSensitive: r = N.isCaseSensitive,
    includeMatches: n = N.includeMatches,
    minMatchCharLength: o = N.minMatchCharLength,
    ignoreLocation: i = N.ignoreLocation,
    findAllMatches: a = N.findAllMatches,
    location: c = N.location,
    threshold: l = N.threshold,
    distance: f = N.distance
  } = {}) {
    this.query = null, this.options = {
      isCaseSensitive: r,
      includeMatches: n,
      minMatchCharLength: o,
      findAllMatches: a,
      ignoreLocation: i,
      location: c,
      threshold: l,
      distance: f
    }, this.pattern = r ? t : t.toLowerCase(), this.query = In(this.pattern, this.options);
  }
  static condition(t, r) {
    return r.useExtendedSearch;
  }
  searchIn(t) {
    const r = this.query;
    if (!r)
      return {
        isMatch: !1,
        score: 1
      };
    const { includeMatches: n, isCaseSensitive: o } = this.options;
    t = o ? t : t.toLowerCase();
    let i = 0, a = [], c = 0;
    for (let l = 0, f = r.length; l < f; l += 1) {
      const d = r[l];
      a.length = 0, i = 0;
      for (let h = 0, y = d.length; h < y; h += 1) {
        const w = d[h], { isMatch: b, indices: A, score: R } = w.search(t);
        if (b) {
          if (i += 1, c += R, n) {
            const O = w.constructor.type;
            Mn.has(O) ? a = [...a, ...A] : a.push(A);
          }
        } else {
          c = 0, i = 0, a.length = 0;
          break;
        }
      }
      if (i) {
        let h = {
          isMatch: !0,
          score: c / i
        };
        return n && (h.indices = a), h;
      }
    }
    return {
      isMatch: !1,
      score: 1
    };
  }
}
const vt = [];
function $n(...e) {
  vt.push(...e);
}
function yt(e, t) {
  for (let r = 0, n = vt.length; r < n; r += 1) {
    let o = vt[r];
    if (o.condition(e, t))
      return new o(e, t);
  }
  return new gr(e, t);
}
const tt = {
  AND: "$and",
  OR: "$or"
}, bt = {
  PATH: "$path",
  PATTERN: "$val"
}, St = (e) => !!(e[tt.AND] || e[tt.OR]), jn = (e) => !!e[bt.PATH], Dn = (e) => !Te(e) && dr(e) && !St(e), Xt = (e) => ({
  [tt.AND]: Object.keys(e).map((t) => ({
    [t]: e[t]
  }))
});
function br(e, t, { auto: r = !0 } = {}) {
  const n = (o) => {
    let i = Object.keys(o);
    const a = jn(o);
    if (!a && i.length > 1 && !St(o))
      return n(Xt(o));
    if (Dn(o)) {
      const l = a ? o[bt.PATH] : i[0], f = a ? o[bt.PATTERN] : o[l];
      if (!_e(f))
        throw new Error(cn(l));
      const d = {
        keyId: mt(l),
        pattern: f
      };
      return r && (d.searcher = yt(f, t)), d;
    }
    let c = {
      children: [],
      operator: i[0]
    };
    return i.forEach((l) => {
      const f = o[l];
      Te(f) && f.forEach((d) => {
        c.children.push(n(d));
      });
    }), c;
  };
  return St(e) || (e = Xt(e)), n(e);
}
function Ln(e, { ignoreFieldNorm: t = N.ignoreFieldNorm }) {
  e.forEach((r) => {
    let n = 1;
    r.matches.forEach(({ key: o, norm: i, score: a }) => {
      const c = o ? o.weight : null;
      n *= Math.pow(
        a === 0 && c ? Number.EPSILON : a,
        (c || 1) * (t ? 1 : i)
      );
    }), r.score = n;
  });
}
function Fn(e, t) {
  const r = e.matches;
  t.matches = [], ge(r) && r.forEach((n) => {
    if (!ge(n.indices) || !n.indices.length)
      return;
    const { indices: o, value: i } = n;
    let a = {
      indices: o,
      value: i
    };
    n.key && (a.key = n.key.src), n.idx > -1 && (a.refIndex = n.idx), t.matches.push(a);
  });
}
function Yn(e, t) {
  t.score = e.score;
}
function zn(e, t, {
  includeMatches: r = N.includeMatches,
  includeScore: n = N.includeScore
} = {}) {
  const o = [];
  return r && o.push(Fn), n && o.push(Yn), e.map((i) => {
    const { idx: a } = i, c = {
      item: t[a],
      refIndex: a
    };
    return o.length && o.forEach((l) => {
      l(i, c);
    }), c;
  });
}
class Ve {
  constructor(t, r = {}, n) {
    this.options = { ...N, ...r }, this.options.useExtendedSearch, this._keyStore = new dn(this.options.keys), this.setCollection(t, n);
  }
  setCollection(t, r) {
    if (this._docs = t, r && !(r instanceof At))
      throw new Error(an);
    this._myIndex = r || mr(this.options.keys, this._docs, {
      getFn: this.options.getFn,
      fieldNormWeight: this.options.fieldNormWeight
    });
  }
  add(t) {
    ge(t) && (this._docs.push(t), this._myIndex.add(t));
  }
  remove(t = () => !1) {
    const r = [];
    for (let n = 0, o = this._docs.length; n < o; n += 1) {
      const i = this._docs[n];
      t(i, n) && (this.removeAt(n), n -= 1, o -= 1, r.push(i));
    }
    return r;
  }
  removeAt(t) {
    this._docs.splice(t, 1), this._myIndex.removeAt(t);
  }
  getIndex() {
    return this._myIndex;
  }
  search(t, { limit: r = -1 } = {}) {
    const {
      includeMatches: n,
      includeScore: o,
      shouldSort: i,
      sortFn: a,
      ignoreFieldNorm: c
    } = this.options;
    let l = _e(t) ? _e(this._docs[0]) ? this._searchStringList(t) : this._searchObjectList(t) : this._searchLogical(t);
    return Ln(l, { ignoreFieldNorm: c }), i && l.sort(a), fr(r) && r > -1 && (l = l.slice(0, r)), zn(l, this._docs, {
      includeMatches: n,
      includeScore: o
    });
  }
  _searchStringList(t) {
    const r = yt(t, this.options), { records: n } = this._myIndex, o = [];
    return n.forEach(({ v: i, i: a, n: c }) => {
      if (!ge(i))
        return;
      const { isMatch: l, score: f, indices: d } = r.searchIn(i);
      l && o.push({
        item: i,
        idx: a,
        matches: [{ score: f, value: i, norm: c, indices: d }]
      });
    }), o;
  }
  _searchLogical(t) {
    const r = br(t, this.options), n = (c, l, f) => {
      if (!c.children) {
        const { keyId: h, searcher: y } = c, w = this._findMatches({
          key: this._keyStore.get(h),
          value: this._myIndex.getValueForItemAtKeyId(l, h),
          searcher: y
        });
        return w && w.length ? [
          {
            idx: f,
            item: l,
            matches: w
          }
        ] : [];
      }
      const d = [];
      for (let h = 0, y = c.children.length; h < y; h += 1) {
        const w = c.children[h], b = n(w, l, f);
        if (b.length)
          d.push(...b);
        else if (c.operator === tt.AND)
          return [];
      }
      return d;
    }, o = this._myIndex.records, i = {}, a = [];
    return o.forEach(({ $: c, i: l }) => {
      if (ge(c)) {
        let f = n(r, c, l);
        f.length && (i[l] || (i[l] = { idx: l, item: c, matches: [] }, a.push(i[l])), f.forEach(({ matches: d }) => {
          i[l].matches.push(...d);
        }));
      }
    }), a;
  }
  _searchObjectList(t) {
    const r = yt(t, this.options), { keys: n, records: o } = this._myIndex, i = [];
    return o.forEach(({ $: a, i: c }) => {
      if (!ge(a))
        return;
      let l = [];
      n.forEach((f, d) => {
        l.push(
          ...this._findMatches({
            key: f,
            value: a[d],
            searcher: r
          })
        );
      }), l.length && i.push({
        idx: c,
        item: a,
        matches: l
      });
    }), i;
  }
  _findMatches({ key: t, value: r, searcher: n }) {
    if (!ge(r))
      return [];
    let o = [];
    if (Te(r))
      r.forEach(({ v: i, i: a, n: c }) => {
        if (!ge(i))
          return;
        const { isMatch: l, score: f, indices: d } = n.searchIn(i);
        l && o.push({
          score: f,
          key: t,
          value: i,
          idx: a,
          norm: c,
          indices: d
        });
      });
    else {
      const { v: i, n: a } = r, { isMatch: c, score: l, indices: f } = n.searchIn(i);
      c && o.push({ score: l, key: t, value: i, norm: a, indices: f });
    }
    return o;
  }
}
Ve.version = "6.6.2";
Ve.createIndex = mr;
Ve.parseIndex = Sn;
Ve.config = N;
Ve.parseQuery = br;
$n(Nn);
var Ye = {}, Vn = {
  get exports() {
    return Ye;
  },
  set exports(e) {
    Ye = e;
  }
}, B = {};
/** @license React v16.13.1
 * react-is.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Jt;
function Wn() {
  if (Jt)
    return B;
  Jt = 1;
  var e = typeof Symbol == "function" && Symbol.for, t = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, a = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, f = e ? Symbol.for("react.concurrent_mode") : 60111, d = e ? Symbol.for("react.forward_ref") : 60112, h = e ? Symbol.for("react.suspense") : 60113, y = e ? Symbol.for("react.suspense_list") : 60120, w = e ? Symbol.for("react.memo") : 60115, b = e ? Symbol.for("react.lazy") : 60116, A = e ? Symbol.for("react.block") : 60121, R = e ? Symbol.for("react.fundamental") : 60117, O = e ? Symbol.for("react.responder") : 60118, _ = e ? Symbol.for("react.scope") : 60119;
  function T(g) {
    if (typeof g == "object" && g !== null) {
      var X = g.$$typeof;
      switch (X) {
        case t:
          switch (g = g.type, g) {
            case l:
            case f:
            case n:
            case i:
            case o:
            case h:
              return g;
            default:
              switch (g = g && g.$$typeof, g) {
                case c:
                case d:
                case b:
                case w:
                case a:
                  return g;
                default:
                  return X;
              }
          }
        case r:
          return X;
      }
    }
  }
  function k(g) {
    return T(g) === f;
  }
  return B.AsyncMode = l, B.ConcurrentMode = f, B.ContextConsumer = c, B.ContextProvider = a, B.Element = t, B.ForwardRef = d, B.Fragment = n, B.Lazy = b, B.Memo = w, B.Portal = r, B.Profiler = i, B.StrictMode = o, B.Suspense = h, B.isAsyncMode = function(g) {
    return k(g) || T(g) === l;
  }, B.isConcurrentMode = k, B.isContextConsumer = function(g) {
    return T(g) === c;
  }, B.isContextProvider = function(g) {
    return T(g) === a;
  }, B.isElement = function(g) {
    return typeof g == "object" && g !== null && g.$$typeof === t;
  }, B.isForwardRef = function(g) {
    return T(g) === d;
  }, B.isFragment = function(g) {
    return T(g) === n;
  }, B.isLazy = function(g) {
    return T(g) === b;
  }, B.isMemo = function(g) {
    return T(g) === w;
  }, B.isPortal = function(g) {
    return T(g) === r;
  }, B.isProfiler = function(g) {
    return T(g) === i;
  }, B.isStrictMode = function(g) {
    return T(g) === o;
  }, B.isSuspense = function(g) {
    return T(g) === h;
  }, B.isValidElementType = function(g) {
    return typeof g == "string" || typeof g == "function" || g === n || g === f || g === i || g === o || g === h || g === y || typeof g == "object" && g !== null && (g.$$typeof === b || g.$$typeof === w || g.$$typeof === a || g.$$typeof === c || g.$$typeof === d || g.$$typeof === R || g.$$typeof === O || g.$$typeof === _ || g.$$typeof === A);
  }, B.typeOf = T, B;
}
var U = {};
/** @license React v16.13.1
 * react-is.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Qt;
function Bn() {
  return Qt || (Qt = 1, process.env.NODE_ENV !== "production" && function() {
    var e = typeof Symbol == "function" && Symbol.for, t = e ? Symbol.for("react.element") : 60103, r = e ? Symbol.for("react.portal") : 60106, n = e ? Symbol.for("react.fragment") : 60107, o = e ? Symbol.for("react.strict_mode") : 60108, i = e ? Symbol.for("react.profiler") : 60114, a = e ? Symbol.for("react.provider") : 60109, c = e ? Symbol.for("react.context") : 60110, l = e ? Symbol.for("react.async_mode") : 60111, f = e ? Symbol.for("react.concurrent_mode") : 60111, d = e ? Symbol.for("react.forward_ref") : 60112, h = e ? Symbol.for("react.suspense") : 60113, y = e ? Symbol.for("react.suspense_list") : 60120, w = e ? Symbol.for("react.memo") : 60115, b = e ? Symbol.for("react.lazy") : 60116, A = e ? Symbol.for("react.block") : 60121, R = e ? Symbol.for("react.fundamental") : 60117, O = e ? Symbol.for("react.responder") : 60118, _ = e ? Symbol.for("react.scope") : 60119;
    function T(p) {
      return typeof p == "string" || typeof p == "function" || // Note: its typeof might be other than 'symbol' or 'number' if it's a polyfill.
      p === n || p === f || p === i || p === o || p === h || p === y || typeof p == "object" && p !== null && (p.$$typeof === b || p.$$typeof === w || p.$$typeof === a || p.$$typeof === c || p.$$typeof === d || p.$$typeof === R || p.$$typeof === O || p.$$typeof === _ || p.$$typeof === A);
    }
    function k(p) {
      if (typeof p == "object" && p !== null) {
        var oe = p.$$typeof;
        switch (oe) {
          case t:
            var se = p.type;
            switch (se) {
              case l:
              case f:
              case n:
              case i:
              case o:
              case h:
                return se;
              default:
                var $ = se && se.$$typeof;
                switch ($) {
                  case c:
                  case d:
                  case b:
                  case w:
                  case a:
                    return $;
                  default:
                    return oe;
                }
            }
          case r:
            return oe;
        }
      }
    }
    var g = l, X = f, J = c, G = a, Q = t, fe = d, de = n, Z = b, q = w, Y = r, H = i, he = o, ne = h, te = !1;
    function ve(p) {
      return te || (te = !0, console.warn("The ReactIs.isAsyncMode() alias has been deprecated, and will be removed in React 17+. Update your code to use ReactIs.isConcurrentMode() instead. It has the exact same API.")), ye(p) || k(p) === l;
    }
    function ye(p) {
      return k(p) === f;
    }
    function Se(p) {
      return k(p) === c;
    }
    function C(p) {
      return k(p) === a;
    }
    function E(p) {
      return typeof p == "object" && p !== null && p.$$typeof === t;
    }
    function x(p) {
      return k(p) === d;
    }
    function M(p) {
      return k(p) === n;
    }
    function u(p) {
      return k(p) === b;
    }
    function D(p) {
      return k(p) === w;
    }
    function v(p) {
      return k(p) === r;
    }
    function K(p) {
      return k(p) === i;
    }
    function j(p) {
      return k(p) === o;
    }
    function F(p) {
      return k(p) === h;
    }
    U.AsyncMode = g, U.ConcurrentMode = X, U.ContextConsumer = J, U.ContextProvider = G, U.Element = Q, U.ForwardRef = fe, U.Fragment = de, U.Lazy = Z, U.Memo = q, U.Portal = Y, U.Profiler = H, U.StrictMode = he, U.Suspense = ne, U.isAsyncMode = ve, U.isConcurrentMode = ye, U.isContextConsumer = Se, U.isContextProvider = C, U.isElement = E, U.isForwardRef = x, U.isFragment = M, U.isLazy = u, U.isMemo = D, U.isPortal = v, U.isProfiler = K, U.isStrictMode = j, U.isSuspense = F, U.isValidElementType = T, U.typeOf = k;
  }()), U;
}
(function(e) {
  process.env.NODE_ENV === "production" ? e.exports = Wn() : e.exports = Bn();
})(Vn);
function Un(e) {
  function t(C, E, x, M, u) {
    for (var D = 0, v = 0, K = 0, j = 0, F, p, oe = 0, se = 0, $, ie = $ = F = 0, V = 0, ce = 0, Ee = 0, ue = 0, je = x.length, Ce = je - 1, me, P = "", re = "", De = "", Re = "", we; V < je; ) {
      if (p = x.charCodeAt(V), V === Ce && v + j + K + D !== 0 && (v !== 0 && (p = v === 47 ? 10 : 47), j = K = D = 0, je++, Ce++), v + j + K + D === 0) {
        if (V === Ce && (0 < ce && (P = P.replace(y, "")), 0 < P.trim().length)) {
          switch (p) {
            case 32:
            case 9:
            case 59:
            case 13:
            case 10:
              break;
            default:
              P += x.charAt(V);
          }
          p = 59;
        }
        switch (p) {
          case 123:
            for (P = P.trim(), F = P.charCodeAt(0), $ = 1, ue = ++V; V < je; ) {
              switch (p = x.charCodeAt(V)) {
                case 123:
                  $++;
                  break;
                case 125:
                  $--;
                  break;
                case 47:
                  switch (p = x.charCodeAt(V + 1)) {
                    case 42:
                    case 47:
                      e: {
                        for (ie = V + 1; ie < Ce; ++ie)
                          switch (x.charCodeAt(ie)) {
                            case 47:
                              if (p === 42 && x.charCodeAt(ie - 1) === 42 && V + 2 !== ie) {
                                V = ie + 1;
                                break e;
                              }
                              break;
                            case 10:
                              if (p === 47) {
                                V = ie + 1;
                                break e;
                              }
                          }
                        V = ie;
                      }
                  }
                  break;
                case 91:
                  p++;
                case 40:
                  p++;
                case 34:
                case 39:
                  for (; V++ < Ce && x.charCodeAt(V) !== p; )
                    ;
              }
              if ($ === 0)
                break;
              V++;
            }
            switch ($ = x.substring(ue, V), F === 0 && (F = (P = P.replace(h, "").trim()).charCodeAt(0)), F) {
              case 64:
                switch (0 < ce && (P = P.replace(y, "")), p = P.charCodeAt(1), p) {
                  case 100:
                  case 109:
                  case 115:
                  case 45:
                    ce = E;
                    break;
                  default:
                    ce = he;
                }
                if ($ = t(E, ce, $, p, u + 1), ue = $.length, 0 < te && (ce = r(he, P, Ee), we = c(3, $, ce, E, q, Z, ue, p, u, M), P = ce.join(""), we !== void 0 && (ue = ($ = we.trim()).length) === 0 && (p = 0, $ = "")), 0 < ue)
                  switch (p) {
                    case 115:
                      P = P.replace(X, a);
                    case 100:
                    case 109:
                    case 45:
                      $ = P + "{" + $ + "}";
                      break;
                    case 107:
                      P = P.replace(_, "$1 $2"), $ = P + "{" + $ + "}", $ = H === 1 || H === 2 && i("@" + $, 3) ? "@-webkit-" + $ + "@" + $ : "@" + $;
                      break;
                    default:
                      $ = P + $, M === 112 && ($ = (re += $, ""));
                  }
                else
                  $ = "";
                break;
              default:
                $ = t(E, r(E, P, Ee), $, M, u + 1);
            }
            De += $, $ = Ee = ce = ie = F = 0, P = "", p = x.charCodeAt(++V);
            break;
          case 125:
          case 59:
            if (P = (0 < ce ? P.replace(y, "") : P).trim(), 1 < (ue = P.length))
              switch (ie === 0 && (F = P.charCodeAt(0), F === 45 || 96 < F && 123 > F) && (ue = (P = P.replace(" ", ":")).length), 0 < te && (we = c(1, P, E, C, q, Z, re.length, M, u, M)) !== void 0 && (ue = (P = we.trim()).length) === 0 && (P = "\0\0"), F = P.charCodeAt(0), p = P.charCodeAt(1), F) {
                case 0:
                  break;
                case 64:
                  if (p === 105 || p === 99) {
                    Re += P + x.charAt(V);
                    break;
                  }
                default:
                  P.charCodeAt(ue - 1) !== 58 && (re += o(P, F, p, P.charCodeAt(2)));
              }
            Ee = ce = ie = F = 0, P = "", p = x.charCodeAt(++V);
        }
      }
      switch (p) {
        case 13:
        case 10:
          v === 47 ? v = 0 : 1 + F === 0 && M !== 107 && 0 < P.length && (ce = 1, P += "\0"), 0 < te * ye && c(0, P, E, C, q, Z, re.length, M, u, M), Z = 1, q++;
          break;
        case 59:
        case 125:
          if (v + j + K + D === 0) {
            Z++;
            break;
          }
        default:
          switch (Z++, me = x.charAt(V), p) {
            case 9:
            case 32:
              if (j + D + v === 0)
                switch (oe) {
                  case 44:
                  case 58:
                  case 9:
                  case 32:
                    me = "";
                    break;
                  default:
                    p !== 32 && (me = " ");
                }
              break;
            case 0:
              me = "\\0";
              break;
            case 12:
              me = "\\f";
              break;
            case 11:
              me = "\\v";
              break;
            case 38:
              j + v + D === 0 && (ce = Ee = 1, me = "\f" + me);
              break;
            case 108:
              if (j + v + D + Y === 0 && 0 < ie)
                switch (V - ie) {
                  case 2:
                    oe === 112 && x.charCodeAt(V - 3) === 58 && (Y = oe);
                  case 8:
                    se === 111 && (Y = se);
                }
              break;
            case 58:
              j + v + D === 0 && (ie = V);
              break;
            case 44:
              v + K + j + D === 0 && (ce = 1, me += "\r");
              break;
            case 34:
            case 39:
              v === 0 && (j = j === p ? 0 : j === 0 ? p : j);
              break;
            case 91:
              j + v + K === 0 && D++;
              break;
            case 93:
              j + v + K === 0 && D--;
              break;
            case 41:
              j + v + D === 0 && K--;
              break;
            case 40:
              if (j + v + D === 0) {
                if (F === 0)
                  switch (2 * oe + 3 * se) {
                    case 533:
                      break;
                    default:
                      F = 1;
                  }
                K++;
              }
              break;
            case 64:
              v + K + j + D + ie + $ === 0 && ($ = 1);
              break;
            case 42:
            case 47:
              if (!(0 < j + D + K))
                switch (v) {
                  case 0:
                    switch (2 * p + 3 * x.charCodeAt(V + 1)) {
                      case 235:
                        v = 47;
                        break;
                      case 220:
                        ue = V, v = 42;
                    }
                    break;
                  case 42:
                    p === 47 && oe === 42 && ue + 2 !== V && (x.charCodeAt(ue + 2) === 33 && (re += x.substring(ue, V + 1)), me = "", v = 0);
                }
          }
          v === 0 && (P += me);
      }
      se = oe, oe = p, V++;
    }
    if (ue = re.length, 0 < ue) {
      if (ce = E, 0 < te && (we = c(2, re, ce, C, q, Z, ue, M, u, M), we !== void 0 && (re = we).length === 0))
        return Re + re + De;
      if (re = ce.join(",") + "{" + re + "}", H * Y !== 0) {
        switch (H !== 2 || i(re, 2) || (Y = 0), Y) {
          case 111:
            re = re.replace(k, ":-moz-$1") + re;
            break;
          case 112:
            re = re.replace(T, "::-webkit-input-$1") + re.replace(T, "::-moz-$1") + re.replace(T, ":-ms-input-$1") + re;
        }
        Y = 0;
      }
    }
    return Re + re + De;
  }
  function r(C, E, x) {
    var M = E.trim().split(R);
    E = M;
    var u = M.length, D = C.length;
    switch (D) {
      case 0:
      case 1:
        var v = 0;
        for (C = D === 0 ? "" : C[0] + " "; v < u; ++v)
          E[v] = n(C, E[v], x).trim();
        break;
      default:
        var K = v = 0;
        for (E = []; v < u; ++v)
          for (var j = 0; j < D; ++j)
            E[K++] = n(C[j] + " ", M[v], x).trim();
    }
    return E;
  }
  function n(C, E, x) {
    var M = E.charCodeAt(0);
    switch (33 > M && (M = (E = E.trim()).charCodeAt(0)), M) {
      case 38:
        return E.replace(O, "$1" + C.trim());
      case 58:
        return C.trim() + E.replace(O, "$1" + C.trim());
      default:
        if (0 < 1 * x && 0 < E.indexOf("\f"))
          return E.replace(O, (C.charCodeAt(0) === 58 ? "" : "$1") + C.trim());
    }
    return C + E;
  }
  function o(C, E, x, M) {
    var u = C + ";", D = 2 * E + 3 * x + 4 * M;
    if (D === 944) {
      C = u.indexOf(":", 9) + 1;
      var v = u.substring(C, u.length - 1).trim();
      return v = u.substring(0, C).trim() + v + ";", H === 1 || H === 2 && i(v, 1) ? "-webkit-" + v + v : v;
    }
    if (H === 0 || H === 2 && !i(u, 1))
      return u;
    switch (D) {
      case 1015:
        return u.charCodeAt(10) === 97 ? "-webkit-" + u + u : u;
      case 951:
        return u.charCodeAt(3) === 116 ? "-webkit-" + u + u : u;
      case 963:
        return u.charCodeAt(5) === 110 ? "-webkit-" + u + u : u;
      case 1009:
        if (u.charCodeAt(4) !== 100)
          break;
      case 969:
      case 942:
        return "-webkit-" + u + u;
      case 978:
        return "-webkit-" + u + "-moz-" + u + u;
      case 1019:
      case 983:
        return "-webkit-" + u + "-moz-" + u + "-ms-" + u + u;
      case 883:
        if (u.charCodeAt(8) === 45)
          return "-webkit-" + u + u;
        if (0 < u.indexOf("image-set(", 11))
          return u.replace(de, "$1-webkit-$2") + u;
        break;
      case 932:
        if (u.charCodeAt(4) === 45)
          switch (u.charCodeAt(5)) {
            case 103:
              return "-webkit-box-" + u.replace("-grow", "") + "-webkit-" + u + "-ms-" + u.replace("grow", "positive") + u;
            case 115:
              return "-webkit-" + u + "-ms-" + u.replace("shrink", "negative") + u;
            case 98:
              return "-webkit-" + u + "-ms-" + u.replace("basis", "preferred-size") + u;
          }
        return "-webkit-" + u + "-ms-" + u + u;
      case 964:
        return "-webkit-" + u + "-ms-flex-" + u + u;
      case 1023:
        if (u.charCodeAt(8) !== 99)
          break;
        return v = u.substring(u.indexOf(":", 15)).replace("flex-", "").replace("space-between", "justify"), "-webkit-box-pack" + v + "-webkit-" + u + "-ms-flex-pack" + v + u;
      case 1005:
        return b.test(u) ? u.replace(w, ":-webkit-") + u.replace(w, ":-moz-") + u : u;
      case 1e3:
        switch (v = u.substring(13).trim(), E = v.indexOf("-") + 1, v.charCodeAt(0) + v.charCodeAt(E)) {
          case 226:
            v = u.replace(g, "tb");
            break;
          case 232:
            v = u.replace(g, "tb-rl");
            break;
          case 220:
            v = u.replace(g, "lr");
            break;
          default:
            return u;
        }
        return "-webkit-" + u + "-ms-" + v + u;
      case 1017:
        if (u.indexOf("sticky", 9) === -1)
          break;
      case 975:
        switch (E = (u = C).length - 10, v = (u.charCodeAt(E) === 33 ? u.substring(0, E) : u).substring(C.indexOf(":", 7) + 1).trim(), D = v.charCodeAt(0) + (v.charCodeAt(7) | 0)) {
          case 203:
            if (111 > v.charCodeAt(8))
              break;
          case 115:
            u = u.replace(v, "-webkit-" + v) + ";" + u;
            break;
          case 207:
          case 102:
            u = u.replace(v, "-webkit-" + (102 < D ? "inline-" : "") + "box") + ";" + u.replace(v, "-webkit-" + v) + ";" + u.replace(v, "-ms-" + v + "box") + ";" + u;
        }
        return u + ";";
      case 938:
        if (u.charCodeAt(5) === 45)
          switch (u.charCodeAt(6)) {
            case 105:
              return v = u.replace("-items", ""), "-webkit-" + u + "-webkit-box-" + v + "-ms-flex-" + v + u;
            case 115:
              return "-webkit-" + u + "-ms-flex-item-" + u.replace(G, "") + u;
            default:
              return "-webkit-" + u + "-ms-flex-line-pack" + u.replace("align-content", "").replace(G, "") + u;
          }
        break;
      case 973:
      case 989:
        if (u.charCodeAt(3) !== 45 || u.charCodeAt(4) === 122)
          break;
      case 931:
      case 953:
        if (fe.test(C) === !0)
          return (v = C.substring(C.indexOf(":") + 1)).charCodeAt(0) === 115 ? o(C.replace("stretch", "fill-available"), E, x, M).replace(":fill-available", ":stretch") : u.replace(v, "-webkit-" + v) + u.replace(v, "-moz-" + v.replace("fill-", "")) + u;
        break;
      case 962:
        if (u = "-webkit-" + u + (u.charCodeAt(5) === 102 ? "-ms-" + u : "") + u, x + M === 211 && u.charCodeAt(13) === 105 && 0 < u.indexOf("transform", 10))
          return u.substring(0, u.indexOf(";", 27) + 1).replace(A, "$1-webkit-$2") + u;
    }
    return u;
  }
  function i(C, E) {
    var x = C.indexOf(E === 1 ? ":" : "{"), M = C.substring(0, E !== 3 ? x : 10);
    return x = C.substring(x + 1, C.length - 1), ve(E !== 2 ? M : M.replace(Q, "$1"), x, E);
  }
  function a(C, E) {
    var x = o(E, E.charCodeAt(0), E.charCodeAt(1), E.charCodeAt(2));
    return x !== E + ";" ? x.replace(J, " or ($1)").substring(4) : "(" + E + ")";
  }
  function c(C, E, x, M, u, D, v, K, j, F) {
    for (var p = 0, oe = E, se; p < te; ++p)
      switch (se = ne[p].call(d, C, oe, x, M, u, D, v, K, j, F)) {
        case void 0:
        case !1:
        case !0:
        case null:
          break;
        default:
          oe = se;
      }
    if (oe !== E)
      return oe;
  }
  function l(C) {
    switch (C) {
      case void 0:
      case null:
        te = ne.length = 0;
        break;
      default:
        if (typeof C == "function")
          ne[te++] = C;
        else if (typeof C == "object")
          for (var E = 0, x = C.length; E < x; ++E)
            l(C[E]);
        else
          ye = !!C | 0;
    }
    return l;
  }
  function f(C) {
    return C = C.prefix, C !== void 0 && (ve = null, C ? typeof C != "function" ? H = 1 : (H = 2, ve = C) : H = 0), f;
  }
  function d(C, E) {
    var x = C;
    if (33 > x.charCodeAt(0) && (x = x.trim()), Se = x, x = [Se], 0 < te) {
      var M = c(-1, E, x, x, q, Z, 0, 0, 0, 0);
      M !== void 0 && typeof M == "string" && (E = M);
    }
    var u = t(he, x, E, 0, 0);
    return 0 < te && (M = c(-2, u, x, x, q, Z, u.length, 0, 0, 0), M !== void 0 && (u = M)), Se = "", Y = 0, Z = q = 1, u;
  }
  var h = /^\0+/g, y = /[\0\r\f]/g, w = /: */g, b = /zoo|gra/, A = /([,: ])(transform)/g, R = /,\r+?/g, O = /([\t\r\n ])*\f?&/g, _ = /@(k\w+)\s*(\S*)\s*/, T = /::(place)/g, k = /:(read-only)/g, g = /[svh]\w+-[tblr]{2}/, X = /\(\s*(.*)\s*\)/g, J = /([\s\S]*?);/g, G = /-self|flex-/g, Q = /[^]*?(:[rp][el]a[\w-]+)[^]*/, fe = /stretch|:\s*\w+\-(?:conte|avail)/, de = /([^-])(image-set\()/, Z = 1, q = 1, Y = 0, H = 1, he = [], ne = [], te = 0, ve = null, ye = 0, Se = "";
  return d.use = l, d.set = f, e !== void 0 && f(e), d;
}
var Gn = {
  animationIterationCount: 1,
  borderImageOutset: 1,
  borderImageSlice: 1,
  borderImageWidth: 1,
  boxFlex: 1,
  boxFlexGroup: 1,
  boxOrdinalGroup: 1,
  columnCount: 1,
  columns: 1,
  flex: 1,
  flexGrow: 1,
  flexPositive: 1,
  flexShrink: 1,
  flexNegative: 1,
  flexOrder: 1,
  gridRow: 1,
  gridRowEnd: 1,
  gridRowSpan: 1,
  gridRowStart: 1,
  gridColumn: 1,
  gridColumnEnd: 1,
  gridColumnSpan: 1,
  gridColumnStart: 1,
  msGridRow: 1,
  msGridRowSpan: 1,
  msGridColumn: 1,
  msGridColumnSpan: 1,
  fontWeight: 1,
  lineHeight: 1,
  opacity: 1,
  order: 1,
  orphans: 1,
  tabSize: 1,
  widows: 1,
  zIndex: 1,
  zoom: 1,
  WebkitLineClamp: 1,
  // SVG-related properties
  fillOpacity: 1,
  floodOpacity: 1,
  stopOpacity: 1,
  strokeDasharray: 1,
  strokeDashoffset: 1,
  strokeMiterlimit: 1,
  strokeOpacity: 1,
  strokeWidth: 1
};
function Hn(e) {
  var t = /* @__PURE__ */ Object.create(null);
  return function(r) {
    return t[r] === void 0 && (t[r] = e(r)), t[r];
  };
}
var Kn = /^((children|dangerouslySetInnerHTML|key|ref|autoFocus|defaultValue|defaultChecked|innerHTML|suppressContentEditableWarning|suppressHydrationWarning|valueLink|abbr|accept|acceptCharset|accessKey|action|allow|allowUserMedia|allowPaymentRequest|allowFullScreen|allowTransparency|alt|async|autoComplete|autoPlay|capture|cellPadding|cellSpacing|challenge|charSet|checked|cite|classID|className|cols|colSpan|content|contentEditable|contextMenu|controls|controlsList|coords|crossOrigin|data|dateTime|decoding|default|defer|dir|disabled|disablePictureInPicture|download|draggable|encType|enterKeyHint|form|formAction|formEncType|formMethod|formNoValidate|formTarget|frameBorder|headers|height|hidden|high|href|hrefLang|htmlFor|httpEquiv|id|inputMode|integrity|is|keyParams|keyType|kind|label|lang|list|loading|loop|low|marginHeight|marginWidth|max|maxLength|media|mediaGroup|method|min|minLength|multiple|muted|name|nonce|noValidate|open|optimum|pattern|placeholder|playsInline|poster|preload|profile|radioGroup|readOnly|referrerPolicy|rel|required|reversed|role|rows|rowSpan|sandbox|scope|scoped|scrolling|seamless|selected|shape|size|sizes|slot|span|spellCheck|src|srcDoc|srcLang|srcSet|start|step|style|summary|tabIndex|target|title|translate|type|useMap|value|width|wmode|wrap|about|datatype|inlist|prefix|property|resource|typeof|vocab|autoCapitalize|autoCorrect|autoSave|color|incremental|fallback|inert|itemProp|itemScope|itemType|itemID|itemRef|on|option|results|security|unselectable|accentHeight|accumulate|additive|alignmentBaseline|allowReorder|alphabetic|amplitude|arabicForm|ascent|attributeName|attributeType|autoReverse|azimuth|baseFrequency|baselineShift|baseProfile|bbox|begin|bias|by|calcMode|capHeight|clip|clipPathUnits|clipPath|clipRule|colorInterpolation|colorInterpolationFilters|colorProfile|colorRendering|contentScriptType|contentStyleType|cursor|cx|cy|d|decelerate|descent|diffuseConstant|direction|display|divisor|dominantBaseline|dur|dx|dy|edgeMode|elevation|enableBackground|end|exponent|externalResourcesRequired|fill|fillOpacity|fillRule|filter|filterRes|filterUnits|floodColor|floodOpacity|focusable|fontFamily|fontSize|fontSizeAdjust|fontStretch|fontStyle|fontVariant|fontWeight|format|from|fr|fx|fy|g1|g2|glyphName|glyphOrientationHorizontal|glyphOrientationVertical|glyphRef|gradientTransform|gradientUnits|hanging|horizAdvX|horizOriginX|ideographic|imageRendering|in|in2|intercept|k|k1|k2|k3|k4|kernelMatrix|kernelUnitLength|kerning|keyPoints|keySplines|keyTimes|lengthAdjust|letterSpacing|lightingColor|limitingConeAngle|local|markerEnd|markerMid|markerStart|markerHeight|markerUnits|markerWidth|mask|maskContentUnits|maskUnits|mathematical|mode|numOctaves|offset|opacity|operator|order|orient|orientation|origin|overflow|overlinePosition|overlineThickness|panose1|paintOrder|pathLength|patternContentUnits|patternTransform|patternUnits|pointerEvents|points|pointsAtX|pointsAtY|pointsAtZ|preserveAlpha|preserveAspectRatio|primitiveUnits|r|radius|refX|refY|renderingIntent|repeatCount|repeatDur|requiredExtensions|requiredFeatures|restart|result|rotate|rx|ry|scale|seed|shapeRendering|slope|spacing|specularConstant|specularExponent|speed|spreadMethod|startOffset|stdDeviation|stemh|stemv|stitchTiles|stopColor|stopOpacity|strikethroughPosition|strikethroughThickness|string|stroke|strokeDasharray|strokeDashoffset|strokeLinecap|strokeLinejoin|strokeMiterlimit|strokeOpacity|strokeWidth|surfaceScale|systemLanguage|tableValues|targetX|targetY|textAnchor|textDecoration|textRendering|textLength|to|transform|u1|u2|underlinePosition|underlineThickness|unicode|unicodeBidi|unicodeRange|unitsPerEm|vAlphabetic|vHanging|vIdeographic|vMathematical|values|vectorEffect|version|vertAdvY|vertOriginX|vertOriginY|viewBox|viewTarget|visibility|widths|wordSpacing|writingMode|x|xHeight|x1|x2|xChannelSelector|xlinkActuate|xlinkArcrole|xlinkHref|xlinkRole|xlinkShow|xlinkTitle|xlinkType|xmlBase|xmlns|xmlnsXlink|xmlLang|xmlSpace|y|y1|y2|yChannelSelector|z|zoomAndPan|for|class|autofocus)|(([Dd][Aa][Tt][Aa]|[Aa][Rr][Ii][Aa]|x)-.*))$/, Zt = /* @__PURE__ */ Hn(
  function(e) {
    return Kn.test(e) || e.charCodeAt(0) === 111 && e.charCodeAt(1) === 110 && e.charCodeAt(2) < 91;
  }
  /* Z+1 */
), kt = Ye, Xn = {
  childContextTypes: !0,
  contextType: !0,
  contextTypes: !0,
  defaultProps: !0,
  displayName: !0,
  getDefaultProps: !0,
  getDerivedStateFromError: !0,
  getDerivedStateFromProps: !0,
  mixins: !0,
  propTypes: !0,
  type: !0
}, Jn = {
  name: !0,
  length: !0,
  prototype: !0,
  caller: !0,
  callee: !0,
  arguments: !0,
  arity: !0
}, Qn = {
  $$typeof: !0,
  render: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0
}, Sr = {
  $$typeof: !0,
  compare: !0,
  defaultProps: !0,
  displayName: !0,
  propTypes: !0,
  type: !0
}, Tt = {};
Tt[kt.ForwardRef] = Qn;
Tt[kt.Memo] = Sr;
function qt(e) {
  return kt.isMemo(e) ? Sr : Tt[e.$$typeof] || Xn;
}
var Zn = Object.defineProperty, qn = Object.getOwnPropertyNames, er = Object.getOwnPropertySymbols, eo = Object.getOwnPropertyDescriptor, to = Object.getPrototypeOf, tr = Object.prototype;
function Er(e, t, r) {
  if (typeof t != "string") {
    if (tr) {
      var n = to(t);
      n && n !== tr && Er(e, n, r);
    }
    var o = qn(t);
    er && (o = o.concat(er(t)));
    for (var i = qt(e), a = qt(t), c = 0; c < o.length; ++c) {
      var l = o[c];
      if (!Jn[l] && !(r && r[l]) && !(a && a[l]) && !(i && i[l])) {
        var f = eo(t, l);
        try {
          Zn(e, l, f);
        } catch {
        }
      }
    }
  }
  return e;
}
var ro = Er;
function xe() {
  return (xe = Object.assign || function(e) {
    for (var t = 1; t < arguments.length; t++) {
      var r = arguments[t];
      for (var n in r)
        Object.prototype.hasOwnProperty.call(r, n) && (e[n] = r[n]);
    }
    return e;
  }).apply(this, arguments);
}
var rr = function(e, t) {
  for (var r = [e[0]], n = 0, o = t.length; n < o; n += 1)
    r.push(t[n], e[n + 1]);
  return r;
}, Et = function(e) {
  return e !== null && typeof e == "object" && (e.toString ? e.toString() : Object.prototype.toString.call(e)) === "[object Object]" && !Ye.typeOf(e);
}, rt = Object.freeze([]), Oe = Object.freeze({});
function He(e) {
  return typeof e == "function";
}
function wt(e) {
  return process.env.NODE_ENV !== "production" && typeof e == "string" && e || e.displayName || e.name || "Component";
}
function Ot(e) {
  return e && typeof e.styledComponentId == "string";
}
var ze = typeof process < "u" && (process.env.REACT_APP_SC_ATTR || process.env.SC_ATTR) || "data-styled", Pt = typeof window < "u" && "HTMLElement" in window, no = Boolean(typeof SC_DISABLE_SPEEDY == "boolean" ? SC_DISABLE_SPEEDY : typeof process < "u" && process.env.REACT_APP_SC_DISABLE_SPEEDY !== void 0 && process.env.REACT_APP_SC_DISABLE_SPEEDY !== "" ? process.env.REACT_APP_SC_DISABLE_SPEEDY !== "false" && process.env.REACT_APP_SC_DISABLE_SPEEDY : typeof process < "u" && process.env.SC_DISABLE_SPEEDY !== void 0 && process.env.SC_DISABLE_SPEEDY !== "" ? process.env.SC_DISABLE_SPEEDY !== "false" && process.env.SC_DISABLE_SPEEDY : process.env.NODE_ENV !== "production"), oo = {}, so = process.env.NODE_ENV !== "production" ? { 1: `Cannot create styled-component for component: %s.

`, 2: `Can't collect styles once you've consumed a \`ServerStyleSheet\`'s styles! \`ServerStyleSheet\` is a one off instance for each server-side render cycle.

- Are you trying to reuse it across renders?
- Are you accidentally calling collectStyles twice?

`, 3: `Streaming SSR is only supported in a Node.js environment; Please do not try to call this method in the browser.

`, 4: `The \`StyleSheetManager\` expects a valid target or sheet prop!

- Does this error occur on the client and is your target falsy?
- Does this error occur on the server and is the sheet falsy?

`, 5: `The clone method cannot be used on the client!

- Are you running in a client-like environment on the server?
- Are you trying to run SSR on the client?

`, 6: `Trying to insert a new style tag, but the given Node is unmounted!

- Are you using a custom target that isn't mounted?
- Does your document not have a valid head element?
- Have you accidentally removed a style tag manually?

`, 7: 'ThemeProvider: Please return an object from your "theme" prop function, e.g.\n\n```js\ntheme={() => ({})}\n```\n\n', 8: `ThemeProvider: Please make your "theme" prop an object.

`, 9: "Missing document `<head>`\n\n", 10: `Cannot find a StyleSheet instance. Usually this happens if there are multiple copies of styled-components loaded at once. Check out this issue for how to troubleshoot and fix the common cases where this situation can happen: https://github.com/styled-components/styled-components/issues/1941#issuecomment-417862021

`, 11: `_This error was replaced with a dev-time warning, it will be deleted for v4 final._ [createGlobalStyle] received children which will not be rendered. Please use the component without passing children elements.

`, 12: "It seems you are interpolating a keyframe declaration (%s) into an untagged string. This was supported in styled-components v3, but is not longer supported in v4 as keyframes are now injected on-demand. Please wrap your string in the css\\`\\` helper which ensures the styles are injected correctly. See https://www.styled-components.com/docs/api#css\n\n", 13: `%s is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.

`, 14: `ThemeProvider: "theme" prop is required.

`, 15: "A stylis plugin has been supplied that is not named. We need a name for each plugin to be able to prevent styling collisions between different stylis configurations within the same app. Before you pass your plugin to `<StyleSheetManager stylisPlugins={[]}>`, please make sure each plugin is uniquely-named, e.g.\n\n```js\nObject.defineProperty(importedPlugin, 'name', { value: 'some-unique-name' });\n```\n\n", 16: `Reached the limit of how many styled components may be created at group %s.
You may only create up to 1,073,741,824 components. If you're creating components dynamically,
as for instance in your render method then you may be running into this limitation.

`, 17: `CSSStyleSheet could not be found on HTMLStyleElement.
Has styled-components' style tag been unmounted or altered by another script?
` } : {};
function io() {
  for (var e = arguments.length <= 0 ? void 0 : arguments[0], t = [], r = 1, n = arguments.length; r < n; r += 1)
    t.push(r < 0 || arguments.length <= r ? void 0 : arguments[r]);
  return t.forEach(function(o) {
    e = e.replace(/%[a-z]/, o);
  }), e;
}
function We(e) {
  for (var t = arguments.length, r = new Array(t > 1 ? t - 1 : 0), n = 1; n < t; n++)
    r[n - 1] = arguments[n];
  throw process.env.NODE_ENV === "production" ? new Error("An error occurred. See https://git.io/JUIaE#" + e + " for more information." + (r.length > 0 ? " Args: " + r.join(", ") : "")) : new Error(io.apply(void 0, [so[e]].concat(r)).trim());
}
var ao = function() {
  function e(r) {
    this.groupSizes = new Uint32Array(512), this.length = 512, this.tag = r;
  }
  var t = e.prototype;
  return t.indexOfGroup = function(r) {
    for (var n = 0, o = 0; o < r; o++)
      n += this.groupSizes[o];
    return n;
  }, t.insertRules = function(r, n) {
    if (r >= this.groupSizes.length) {
      for (var o = this.groupSizes, i = o.length, a = i; r >= a; )
        (a <<= 1) < 0 && We(16, "" + r);
      this.groupSizes = new Uint32Array(a), this.groupSizes.set(o), this.length = a;
      for (var c = i; c < a; c++)
        this.groupSizes[c] = 0;
    }
    for (var l = this.indexOfGroup(r + 1), f = 0, d = n.length; f < d; f++)
      this.tag.insertRule(l, n[f]) && (this.groupSizes[r]++, l++);
  }, t.clearGroup = function(r) {
    if (r < this.length) {
      var n = this.groupSizes[r], o = this.indexOfGroup(r), i = o + n;
      this.groupSizes[r] = 0;
      for (var a = o; a < i; a++)
        this.tag.deleteRule(o);
    }
  }, t.getGroup = function(r) {
    var n = "";
    if (r >= this.length || this.groupSizes[r] === 0)
      return n;
    for (var o = this.groupSizes[r], i = this.indexOfGroup(r), a = i + o, c = i; c < a; c++)
      n += this.tag.getRule(c) + `/*!sc*/
`;
    return n;
  }, e;
}(), qe = /* @__PURE__ */ new Map(), nt = /* @__PURE__ */ new Map(), Ge = 1, Je = function(e) {
  if (qe.has(e))
    return qe.get(e);
  for (; nt.has(Ge); )
    Ge++;
  var t = Ge++;
  return process.env.NODE_ENV !== "production" && ((0 | t) < 0 || t > 1 << 30) && We(16, "" + t), qe.set(e, t), nt.set(t, e), t;
}, co = function(e) {
  return nt.get(e);
}, lo = function(e, t) {
  t >= Ge && (Ge = t + 1), qe.set(e, t), nt.set(t, e);
}, uo = "style[" + ze + '][data-styled-version="5.3.6"]', fo = new RegExp("^" + ze + '\\.g(\\d+)\\[id="([\\w\\d-]+)"\\].*?"([^"]*)'), ho = function(e, t, r) {
  for (var n, o = r.split(","), i = 0, a = o.length; i < a; i++)
    (n = o[i]) && e.registerName(t, n);
}, po = function(e, t) {
  for (var r = (t.textContent || "").split(`/*!sc*/
`), n = [], o = 0, i = r.length; o < i; o++) {
    var a = r[o].trim();
    if (a) {
      var c = a.match(fo);
      if (c) {
        var l = 0 | parseInt(c[1], 10), f = c[2];
        l !== 0 && (lo(f, l), ho(e, f, c[3]), e.getTag().insertRules(l, n)), n.length = 0;
      } else
        n.push(a);
    }
  }
}, mo = function() {
  return typeof __webpack_nonce__ < "u" ? __webpack_nonce__ : null;
}, wr = function(e) {
  var t = document.head, r = e || t, n = document.createElement("style"), o = function(c) {
    for (var l = c.childNodes, f = l.length; f >= 0; f--) {
      var d = l[f];
      if (d && d.nodeType === 1 && d.hasAttribute(ze))
        return d;
    }
  }(r), i = o !== void 0 ? o.nextSibling : null;
  n.setAttribute(ze, "active"), n.setAttribute("data-styled-version", "5.3.6");
  var a = mo();
  return a && n.setAttribute("nonce", a), r.insertBefore(n, i), n;
}, go = function() {
  function e(r) {
    var n = this.element = wr(r);
    n.appendChild(document.createTextNode("")), this.sheet = function(o) {
      if (o.sheet)
        return o.sheet;
      for (var i = document.styleSheets, a = 0, c = i.length; a < c; a++) {
        var l = i[a];
        if (l.ownerNode === o)
          return l;
      }
      We(17);
    }(n), this.length = 0;
  }
  var t = e.prototype;
  return t.insertRule = function(r, n) {
    try {
      return this.sheet.insertRule(n, r), this.length++, !0;
    } catch {
      return !1;
    }
  }, t.deleteRule = function(r) {
    this.sheet.deleteRule(r), this.length--;
  }, t.getRule = function(r) {
    var n = this.sheet.cssRules[r];
    return n !== void 0 && typeof n.cssText == "string" ? n.cssText : "";
  }, e;
}(), vo = function() {
  function e(r) {
    var n = this.element = wr(r);
    this.nodes = n.childNodes, this.length = 0;
  }
  var t = e.prototype;
  return t.insertRule = function(r, n) {
    if (r <= this.length && r >= 0) {
      var o = document.createTextNode(n), i = this.nodes[r];
      return this.element.insertBefore(o, i || null), this.length++, !0;
    }
    return !1;
  }, t.deleteRule = function(r) {
    this.element.removeChild(this.nodes[r]), this.length--;
  }, t.getRule = function(r) {
    return r < this.length ? this.nodes[r].textContent : "";
  }, e;
}(), yo = function() {
  function e(r) {
    this.rules = [], this.length = 0;
  }
  var t = e.prototype;
  return t.insertRule = function(r, n) {
    return r <= this.length && (this.rules.splice(r, 0, n), this.length++, !0);
  }, t.deleteRule = function(r) {
    this.rules.splice(r, 1), this.length--;
  }, t.getRule = function(r) {
    return r < this.length ? this.rules[r] : "";
  }, e;
}(), nr = Pt, bo = { isServer: !Pt, useCSSOMInjection: !no }, ot = function() {
  function e(r, n, o) {
    r === void 0 && (r = Oe), n === void 0 && (n = {}), this.options = xe({}, bo, {}, r), this.gs = n, this.names = new Map(o), this.server = !!r.isServer, !this.server && Pt && nr && (nr = !1, function(i) {
      for (var a = document.querySelectorAll(uo), c = 0, l = a.length; c < l; c++) {
        var f = a[c];
        f && f.getAttribute(ze) !== "active" && (po(i, f), f.parentNode && f.parentNode.removeChild(f));
      }
    }(this));
  }
  e.registerId = function(r) {
    return Je(r);
  };
  var t = e.prototype;
  return t.reconstructWithOptions = function(r, n) {
    return n === void 0 && (n = !0), new e(xe({}, this.options, {}, r), this.gs, n && this.names || void 0);
  }, t.allocateGSInstance = function(r) {
    return this.gs[r] = (this.gs[r] || 0) + 1;
  }, t.getTag = function() {
    return this.tag || (this.tag = (o = (n = this.options).isServer, i = n.useCSSOMInjection, a = n.target, r = o ? new yo(a) : i ? new go(a) : new vo(a), new ao(r)));
    var r, n, o, i, a;
  }, t.hasNameForId = function(r, n) {
    return this.names.has(r) && this.names.get(r).has(n);
  }, t.registerName = function(r, n) {
    if (Je(r), this.names.has(r))
      this.names.get(r).add(n);
    else {
      var o = /* @__PURE__ */ new Set();
      o.add(n), this.names.set(r, o);
    }
  }, t.insertRules = function(r, n, o) {
    this.registerName(r, n), this.getTag().insertRules(Je(r), o);
  }, t.clearNames = function(r) {
    this.names.has(r) && this.names.get(r).clear();
  }, t.clearRules = function(r) {
    this.getTag().clearGroup(Je(r)), this.clearNames(r);
  }, t.clearTag = function() {
    this.tag = void 0;
  }, t.toString = function() {
    return function(r) {
      for (var n = r.getTag(), o = n.length, i = "", a = 0; a < o; a++) {
        var c = co(a);
        if (c !== void 0) {
          var l = r.names.get(c), f = n.getGroup(a);
          if (l && f && l.size) {
            var d = ze + ".g" + a + '[id="' + c + '"]', h = "";
            l !== void 0 && l.forEach(function(y) {
              y.length > 0 && (h += y + ",");
            }), i += "" + f + d + '{content:"' + h + `"}/*!sc*/
`;
          }
        }
      }
      return i;
    }(this);
  }, e;
}(), So = /(a)(d)/gi, or = function(e) {
  return String.fromCharCode(e + (e > 25 ? 39 : 97));
};
function _t(e) {
  var t, r = "";
  for (t = Math.abs(e); t > 52; t = t / 52 | 0)
    r = or(t % 52) + r;
  return (or(t % 52) + r).replace(So, "$1-$2");
}
var Ne = function(e, t) {
  for (var r = t.length; r; )
    e = 33 * e ^ t.charCodeAt(--r);
  return e;
}, _r = function(e) {
  return Ne(5381, e);
};
function xr(e) {
  for (var t = 0; t < e.length; t += 1) {
    var r = e[t];
    if (He(r) && !Ot(r))
      return !1;
  }
  return !0;
}
var Eo = _r("5.3.6"), wo = function() {
  function e(t, r, n) {
    this.rules = t, this.staticRulesId = "", this.isStatic = process.env.NODE_ENV === "production" && (n === void 0 || n.isStatic) && xr(t), this.componentId = r, this.baseHash = Ne(Eo, r), this.baseStyle = n, ot.registerId(r);
  }
  return e.prototype.generateAndInjectStyles = function(t, r, n) {
    var o = this.componentId, i = [];
    if (this.baseStyle && i.push(this.baseStyle.generateAndInjectStyles(t, r, n)), this.isStatic && !n.hash)
      if (this.staticRulesId && r.hasNameForId(o, this.staticRulesId))
        i.push(this.staticRulesId);
      else {
        var a = $e(this.rules, t, r, n).join(""), c = _t(Ne(this.baseHash, a) >>> 0);
        if (!r.hasNameForId(o, c)) {
          var l = n(a, "." + c, void 0, o);
          r.insertRules(o, c, l);
        }
        i.push(c), this.staticRulesId = c;
      }
    else {
      for (var f = this.rules.length, d = Ne(this.baseHash, n.hash), h = "", y = 0; y < f; y++) {
        var w = this.rules[y];
        if (typeof w == "string")
          h += w, process.env.NODE_ENV !== "production" && (d = Ne(d, w + y));
        else if (w) {
          var b = $e(w, t, r, n), A = Array.isArray(b) ? b.join("") : b;
          d = Ne(d, A + y), h += A;
        }
      }
      if (h) {
        var R = _t(d >>> 0);
        if (!r.hasNameForId(o, R)) {
          var O = n(h, "." + R, void 0, o);
          r.insertRules(o, R, O);
        }
        i.push(R);
      }
    }
    return i.join(" ");
  }, e;
}(), _o = /^\s*\/\/.*$/gm, xo = [":", "[", ".", "#"];
function Co(e) {
  var t, r, n, o, i = e === void 0 ? Oe : e, a = i.options, c = a === void 0 ? Oe : a, l = i.plugins, f = l === void 0 ? rt : l, d = new Un(c), h = [], y = function(A) {
    function R(O) {
      if (O)
        try {
          A(O + "}");
        } catch {
        }
    }
    return function(O, _, T, k, g, X, J, G, Q, fe) {
      switch (O) {
        case 1:
          if (Q === 0 && _.charCodeAt(0) === 64)
            return A(_ + ";"), "";
          break;
        case 2:
          if (G === 0)
            return _ + "/*|*/";
          break;
        case 3:
          switch (G) {
            case 102:
            case 112:
              return A(T[0] + _), "";
            default:
              return _ + (fe === 0 ? "/*|*/" : "");
          }
        case -2:
          _.split("/*|*/}").forEach(R);
      }
    };
  }(function(A) {
    h.push(A);
  }), w = function(A, R, O) {
    return R === 0 && xo.indexOf(O[r.length]) !== -1 || O.match(o) ? A : "." + t;
  };
  function b(A, R, O, _) {
    _ === void 0 && (_ = "&");
    var T = A.replace(_o, ""), k = R && O ? O + " " + R + " { " + T + " }" : T;
    return t = _, r = R, n = new RegExp("\\" + r + "\\b", "g"), o = new RegExp("(\\" + r + "\\b){2,}"), d(O || !R ? "" : R, k);
  }
  return d.use([].concat(f, [function(A, R, O) {
    A === 2 && O.length && O[0].lastIndexOf(r) > 0 && (O[0] = O[0].replace(n, w));
  }, y, function(A) {
    if (A === -2) {
      var R = h;
      return h = [], R;
    }
  }])), b.hash = f.length ? f.reduce(function(A, R) {
    return R.name || We(15), Ne(A, R.name);
  }, 5381).toString() : "", b;
}
var Cr = Pe.createContext();
Cr.Consumer;
var Rr = Pe.createContext(), Ro = (Rr.Consumer, new ot()), xt = Co();
function Ar() {
  return it(Cr) || Ro;
}
function kr() {
  return it(Rr) || xt;
}
var Ao = function() {
  function e(t, r) {
    var n = this;
    this.inject = function(o, i) {
      i === void 0 && (i = xt);
      var a = n.name + i.hash;
      o.hasNameForId(n.id, a) || o.insertRules(n.id, a, i(n.rules, a, "@keyframes"));
    }, this.toString = function() {
      return We(12, String(n.name));
    }, this.name = t, this.id = "sc-keyframes-" + t, this.rules = r;
  }
  return e.prototype.getName = function(t) {
    return t === void 0 && (t = xt), this.name + t.hash;
  }, e;
}(), ko = /([A-Z])/, To = /([A-Z])/g, Oo = /^ms-/, Po = function(e) {
  return "-" + e.toLowerCase();
};
function sr(e) {
  return ko.test(e) ? e.replace(To, Po).replace(Oo, "-ms-") : e;
}
var ir = function(e) {
  return e == null || e === !1 || e === "";
};
function $e(e, t, r, n) {
  if (Array.isArray(e)) {
    for (var o, i = [], a = 0, c = e.length; a < c; a += 1)
      (o = $e(e[a], t, r, n)) !== "" && (Array.isArray(o) ? i.push.apply(i, o) : i.push(o));
    return i;
  }
  if (ir(e))
    return "";
  if (Ot(e))
    return "." + e.styledComponentId;
  if (He(e)) {
    if (typeof (f = e) != "function" || f.prototype && f.prototype.isReactComponent || !t)
      return e;
    var l = e(t);
    return process.env.NODE_ENV !== "production" && Ye.isElement(l) && console.warn(wt(e) + " is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details."), $e(l, t, r, n);
  }
  var f;
  return e instanceof Ao ? r ? (e.inject(r, n), e.getName(n)) : e : Et(e) ? function d(h, y) {
    var w, b, A = [];
    for (var R in h)
      h.hasOwnProperty(R) && !ir(h[R]) && (Array.isArray(h[R]) && h[R].isCss || He(h[R]) ? A.push(sr(R) + ":", h[R], ";") : Et(h[R]) ? A.push.apply(A, d(h[R], R)) : A.push(sr(R) + ": " + (w = R, (b = h[R]) == null || typeof b == "boolean" || b === "" ? "" : typeof b != "number" || b === 0 || w in Gn ? String(b).trim() : b + "px") + ";"));
    return y ? [y + " {"].concat(A, ["}"]) : A;
  }(e) : e.toString();
}
var ar = function(e) {
  return Array.isArray(e) && (e.isCss = !0), e;
};
function Tr(e) {
  for (var t = arguments.length, r = new Array(t > 1 ? t - 1 : 0), n = 1; n < t; n++)
    r[n - 1] = arguments[n];
  return He(e) || Et(e) ? ar($e(rr(rt, [e].concat(r)))) : r.length === 0 && e.length === 1 && typeof e[0] == "string" ? e : ar($e(rr(e, r)));
}
var cr = /invalid hook call/i, Qe = /* @__PURE__ */ new Set(), Or = function(e, t) {
  if (process.env.NODE_ENV !== "production") {
    var r = "The component " + e + (t ? ' with the id of "' + t + '"' : "") + ` has been created dynamically.
You may see this warning because you've called styled inside another component.
To resolve this only create new StyledComponents outside of any render method and function component.`, n = console.error;
    try {
      var o = !0;
      console.error = function(i) {
        if (cr.test(i))
          o = !1, Qe.delete(r);
        else {
          for (var a = arguments.length, c = new Array(a > 1 ? a - 1 : 0), l = 1; l < a; l++)
            c[l - 1] = arguments[l];
          n.apply(void 0, [i].concat(c));
        }
      }, st(), o && !Qe.has(r) && (console.warn(r), Qe.add(r));
    } catch (i) {
      cr.test(i.message) && Qe.delete(r);
    } finally {
      console.error = n;
    }
  }
}, Pr = function(e, t, r) {
  return r === void 0 && (r = Oe), e.theme !== r.theme && e.theme || t || r.theme;
}, Io = /[!"#$%&'()*+,./:;<=>?@[\\\]^`{|}~-]+/g, Mo = /(^-|-$)/g;
function ht(e) {
  return e.replace(Io, "-").replace(Mo, "");
}
var Ir = function(e) {
  return _t(_r(e) >>> 0);
};
function Ze(e) {
  return typeof e == "string" && (process.env.NODE_ENV === "production" || e.charAt(0) === e.charAt(0).toLowerCase());
}
var Ct = function(e) {
  return typeof e == "function" || typeof e == "object" && e !== null && !Array.isArray(e);
}, No = function(e) {
  return e !== "__proto__" && e !== "constructor" && e !== "prototype";
};
function $o(e, t, r) {
  var n = e[r];
  Ct(t) && Ct(n) ? Mr(n, t) : e[r] = t;
}
function Mr(e) {
  for (var t = arguments.length, r = new Array(t > 1 ? t - 1 : 0), n = 1; n < t; n++)
    r[n - 1] = arguments[n];
  for (var o = 0, i = r; o < i.length; o++) {
    var a = i[o];
    if (Ct(a))
      for (var c in a)
        No(c) && $o(e, a[c], c);
  }
  return e;
}
var It = Pe.createContext();
It.Consumer;
var pt = {};
function Nr(e, t, r) {
  var n = Ot(e), o = !Ze(e), i = t.attrs, a = i === void 0 ? rt : i, c = t.componentId, l = c === void 0 ? function(_, T) {
    var k = typeof _ != "string" ? "sc" : ht(_);
    pt[k] = (pt[k] || 0) + 1;
    var g = k + "-" + Ir("5.3.6" + k + pt[k]);
    return T ? T + "-" + g : g;
  }(t.displayName, t.parentComponentId) : c, f = t.displayName, d = f === void 0 ? function(_) {
    return Ze(_) ? "styled." + _ : "Styled(" + wt(_) + ")";
  }(e) : f, h = t.displayName && t.componentId ? ht(t.displayName) + "-" + t.componentId : t.componentId || l, y = n && e.attrs ? Array.prototype.concat(e.attrs, a).filter(Boolean) : a, w = t.shouldForwardProp;
  n && e.shouldForwardProp && (w = t.shouldForwardProp ? function(_, T, k) {
    return e.shouldForwardProp(_, T, k) && t.shouldForwardProp(_, T, k);
  } : e.shouldForwardProp);
  var b, A = new wo(r, h, n ? e.componentStyle : void 0), R = A.isStatic && a.length === 0, O = function(_, T) {
    return function(k, g, X, J) {
      var G = k.attrs, Q = k.componentStyle, fe = k.defaultProps, de = k.foldedComponentIds, Z = k.shouldForwardProp, q = k.styledComponentId, Y = k.target;
      process.env.NODE_ENV !== "production" && Vt(q);
      var H = function(M, u, D) {
        M === void 0 && (M = Oe);
        var v = xe({}, u, { theme: M }), K = {};
        return D.forEach(function(j) {
          var F, p, oe, se = j;
          for (F in He(se) && (se = se(v)), se)
            v[F] = K[F] = F === "className" ? (p = K[F], oe = se[F], p && oe ? p + " " + oe : p || oe) : se[F];
        }), [v, K];
      }(Pr(g, it(It), fe) || Oe, g, G), he = H[0], ne = H[1], te = function(M, u, D, v) {
        var K = Ar(), j = kr(), F = u ? M.generateAndInjectStyles(Oe, K, j) : M.generateAndInjectStyles(D, K, j);
        return process.env.NODE_ENV !== "production" && Vt(F), process.env.NODE_ENV !== "production" && !u && v && v(F), F;
      }(Q, J, he, process.env.NODE_ENV !== "production" ? k.warnTooManyClasses : void 0), ve = X, ye = ne.$as || g.$as || ne.as || g.as || Y, Se = Ze(ye), C = ne !== g ? xe({}, g, {}, ne) : g, E = {};
      for (var x in C)
        x[0] !== "$" && x !== "as" && (x === "forwardedAs" ? E.as = C[x] : (Z ? Z(x, Zt, ye) : !Se || Zt(x)) && (E[x] = C[x]));
      return g.style && ne.style !== g.style && (E.style = xe({}, g.style, {}, ne.style)), E.className = Array.prototype.concat(de, q, te !== q ? te : null, g.className, ne.className).filter(Boolean).join(" "), E.ref = ve, Qr(ye, E);
    }(b, _, T, R);
  };
  return O.displayName = d, (b = Pe.forwardRef(O)).attrs = y, b.componentStyle = A, b.displayName = d, b.shouldForwardProp = w, b.foldedComponentIds = n ? Array.prototype.concat(e.foldedComponentIds, e.styledComponentId) : rt, b.styledComponentId = h, b.target = n ? e.target : e, b.withComponent = function(_) {
    var T = t.componentId, k = function(X, J) {
      if (X == null)
        return {};
      var G, Q, fe = {}, de = Object.keys(X);
      for (Q = 0; Q < de.length; Q++)
        G = de[Q], J.indexOf(G) >= 0 || (fe[G] = X[G]);
      return fe;
    }(t, ["componentId"]), g = T && T + "-" + (Ze(_) ? _ : ht(wt(_)));
    return Nr(_, xe({}, k, { attrs: y, componentId: g }), r);
  }, Object.defineProperty(b, "defaultProps", { get: function() {
    return this._foldedDefaultProps;
  }, set: function(_) {
    this._foldedDefaultProps = n ? Mr({}, e.defaultProps, _) : _;
  } }), process.env.NODE_ENV !== "production" && (Or(d, h), b.warnTooManyClasses = function(_, T) {
    var k = {}, g = !1;
    return function(X) {
      if (!g && (k[X] = !0, Object.keys(k).length >= 200)) {
        var J = T ? ' with the id of "' + T + '"' : "";
        console.warn("Over 200 classes were generated for component " + _ + J + `.
Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))\`width: 100%;\`

  <Component />`), g = !0, k = {};
      }
    };
  }(d, h)), b.toString = function() {
    return "." + b.styledComponentId;
  }, o && ro(b, e, { attrs: !0, componentStyle: !0, displayName: !0, foldedComponentIds: !0, shouldForwardProp: !0, styledComponentId: !0, target: !0, withComponent: !0 }), b;
}
var Rt = function(e) {
  return function t(r, n, o) {
    if (o === void 0 && (o = Oe), !Ye.isValidElementType(n))
      return We(1, String(n));
    var i = function() {
      return r(n, o, Tr.apply(void 0, arguments));
    };
    return i.withConfig = function(a) {
      return t(r, n, xe({}, o, {}, a));
    }, i.attrs = function(a) {
      return t(r, n, xe({}, o, { attrs: Array.prototype.concat(o.attrs, a).filter(Boolean) }));
    }, i;
  }(Nr, e);
};
["a", "abbr", "address", "area", "article", "aside", "audio", "b", "base", "bdi", "bdo", "big", "blockquote", "body", "br", "button", "canvas", "caption", "cite", "code", "col", "colgroup", "data", "datalist", "dd", "del", "details", "dfn", "dialog", "div", "dl", "dt", "em", "embed", "fieldset", "figcaption", "figure", "footer", "form", "h1", "h2", "h3", "h4", "h5", "h6", "head", "header", "hgroup", "hr", "html", "i", "iframe", "img", "input", "ins", "kbd", "keygen", "label", "legend", "li", "link", "main", "map", "mark", "marquee", "menu", "menuitem", "meta", "meter", "nav", "noscript", "object", "ol", "optgroup", "option", "output", "p", "param", "picture", "pre", "progress", "q", "rp", "rt", "ruby", "s", "samp", "script", "section", "select", "small", "source", "span", "strong", "style", "sub", "summary", "sup", "table", "tbody", "td", "textarea", "tfoot", "th", "thead", "time", "title", "tr", "track", "u", "ul", "var", "video", "wbr", "circle", "clipPath", "defs", "ellipse", "foreignObject", "g", "image", "line", "linearGradient", "marker", "mask", "path", "pattern", "polygon", "polyline", "radialGradient", "rect", "stop", "svg", "text", "textPath", "tspan"].forEach(function(e) {
  Rt[e] = Rt(e);
});
var jo = function() {
  function e(r, n) {
    this.rules = r, this.componentId = n, this.isStatic = xr(r), ot.registerId(this.componentId + 1);
  }
  var t = e.prototype;
  return t.createStyles = function(r, n, o, i) {
    var a = i($e(this.rules, n, o, i).join(""), ""), c = this.componentId + r;
    o.insertRules(c, c, a);
  }, t.removeStyles = function(r, n) {
    n.clearRules(this.componentId + r);
  }, t.renderStyles = function(r, n, o, i) {
    r > 2 && ot.registerId(this.componentId + r), this.removeStyles(r, o), this.createStyles(r, n, o, i);
  }, e;
}();
function Do(e) {
  for (var t = arguments.length, r = new Array(t > 1 ? t - 1 : 0), n = 1; n < t; n++)
    r[n - 1] = arguments[n];
  var o = Tr.apply(void 0, [e].concat(r)), i = "sc-global-" + Ir(JSON.stringify(o)), a = new jo(o, i);
  function c(f) {
    var d = Ar(), h = kr(), y = it(It), w = st(d.allocateGSInstance(i)).current;
    return process.env.NODE_ENV !== "production" && Pe.Children.count(f.children) && console.warn("The global style component " + i + " was given child JSX. createGlobalStyle does not render children."), process.env.NODE_ENV !== "production" && o.some(function(b) {
      return typeof b == "string" && b.indexOf("@import") !== -1;
    }) && console.warn("Please do not use @import CSS syntax in createGlobalStyle at this time, as the CSSOM APIs we use in production do not handle it well. Instead, we recommend using a library such as react-helmet to inject a typical <link> meta tag to the stylesheet, or simply embedding it manually in your index.html <head> section for a simpler app."), d.server && l(w, f, d, y, h), Jr(function() {
      if (!d.server)
        return l(w, f, d, y, h), function() {
          return a.removeStyles(w, d);
        };
    }, [w, f, d, y, h]), null;
  }
  function l(f, d, h, y, w) {
    if (a.isStatic)
      a.renderStyles(f, oo, h, w);
    else {
      var b = xe({}, d, { theme: Pr(d, y, c.defaultProps) });
      a.renderStyles(f, b, h, w);
    }
  }
  return process.env.NODE_ENV !== "production" && Or(i), Pe.memo(c);
}
process.env.NODE_ENV !== "production" && typeof navigator < "u" && navigator.product === "ReactNative" && console.warn(`It looks like you've imported 'styled-components' on React Native.
Perhaps you're looking to import 'styled-components/native'?
Read more about this at https://www.styled-components.com/docs/basics#react-native`), process.env.NODE_ENV !== "production" && process.env.NODE_ENV !== "test" && typeof window < "u" && (window["__styled-components-init__"] = window["__styled-components-init__"] || 0, window["__styled-components-init__"] === 1 && console.warn(`It looks like there are several instances of 'styled-components' initialized in this application. This may cause dynamic styles to not render properly, errors during the rehydration process, a missing theme prop, and makes your application bigger without good reason.

See https://s-c.sh/2BAXzed for more info.`), window["__styled-components-init__"] += 1);
const Ke = Rt, Lo = Ke.div`
  margin: 10px 0 0;
  background-color: #fff;
  box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 0, rgba(0, 0, 0, 0.1) 0px 4px 11px;
  border-radius: 5px;
  > ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
    > li {
      padding: 10px 20px;
      background-color: #fff;
      border: 1px solid
        ${({
  dropdownBorderColor: e
}) => e};
      height: 40px;
      display: flex;
      align-items: center;
      &:hover {
        background-color: ${({ dropdownHoverColor: e }) => e};
        cursor: pointer;
      }
      &:first-child {
        border-top-left-radius: 5px;
        border-top-right-radius: 5px;
      }
      &:last-child {
        border-bottom-left-radius: 5px;
        border-bottom-right-radius: 5px;
      }
      &:not(:first-child) {
        border-top: 0;
      }
    }
  }
`, Fo = ({
  onClick: e,
  matchedRecords: t = [],
  dropdownBorderColor: r,
  dropdownHoverColor: n
}) => /* @__PURE__ */ ke(
  Lo,
  {
    className: "react-search-box-dropdown",
    dropdownHoverColor: n,
    dropdownBorderColor: r,
    children: /* @__PURE__ */ ke("ul", { children: t.map((o) => /* @__PURE__ */ ke(
      "li",
      {
        tabIndex: 0,
        onKeyDown: (i) => {
          ["Space", "Enter"].includes(i.code) && e(o);
        },
        className: "react-search-box-dropdown-list-item",
        onClick: () => e(o),
        children: o.item.value
      },
      o.item.key
    )) })
  }
), Yo = (e, t, r) => {
  lr(() => {
    const n = (o) => {
      e.current && !e.current.contains(o.target) && (t(!1), r(""));
    };
    return document.addEventListener("mousedown", n), () => {
      document.removeEventListener("mousedown", n);
    };
  }, [e]);
}, zo = Ke.input`
  font-size: ${({
  inputFontSize: e
}) => e};
  padding: ${({ leftIcon: e, iconBoxSize: t }) => e ? `10px ${t}` : "10px 20px"};
  height: ${({ inputHeight: e }) => e};
  border: 1px solid ${({ inputBorderColor: e }) => e};
  border-radius: 5px;
  color: ${({ inputFontColor: e }) => e};
  background-color: ${({ inputBackgroundColor: e }) => e};
  width: 100%;

  &:focus {
    outline: none;
  }
`, Vo = Ke.span`
  height: ${({
  inputHeight: e
}) => e};
  width: ${({ iconBoxSize: e }) => e};
  position: absolute;
  left: 0;
  top: 0;
  display: flex;
  justify-content: center;
  align-items: center;
`, Wo = Ke.span`
  position: relative;
`, Bo = ({
  placeholder: e,
  name: t,
  value: r,
  onChange: n,
  onFocus: o,
  inputFontColor: i,
  inputBorderColor: a,
  inputFontSize: c,
  inputHeight: l,
  inputBackgroundColor: f,
  autoFocus: d,
  leftIcon: h,
  iconBoxSize: y,
  type: w,
  onKeyDown: b
}) => {
  const A = st(null);
  return lr(() => {
    var O;
    d && ((O = A.current) == null || O.focus());
  }, []), /* @__PURE__ */ ur(Wo, { children: [
    /* @__PURE__ */ ke(
      zo,
      {
        type: w,
        placeholder: e,
        name: t,
        value: r,
        onChange: n,
        onFocus: o,
        ref: A,
        inputFontColor: i,
        inputBorderColor: a,
        inputFontSize: c,
        inputHeight: l,
        inputBackgroundColor: f,
        leftIcon: h,
        iconBoxSize: y,
        onKeyDown: b
      }
    ),
    (() => h ? /* @__PURE__ */ ke(Vo, { iconBoxSize: y, inputHeight: l, children: h }) : null)()
  ] });
}, Uo = Do`
  * {
    box-sizing: border-box;
  }
`, Go = Ke.div`
  display: flex;
  flex-direction: column;
`, Ko = ({
  placeholder: e = "",
  name: t = "",
  data: r = [],
  fuseConfigs: n,
  autoFocus: o = !1,
  onSelect: i,
  onFocus: a,
  onChange: c,
  inputBackgroundColor: l = "#fff",
  inputFontColor: f = "#000",
  inputBorderColor: d = "#cacaca96",
  inputFontSize: h = "14px",
  inputHeight: y = "40px",
  dropdownHoverColor: w = "#ccc",
  dropdownBorderColor: b = "#cacaca96",
  clearOnSelect: A = !1,
  leftIcon: R,
  iconBoxSize: O = "24px",
  type: _ = "text"
}) => {
  const [T, k] = ft([]), [g, X] = ft(""), [J, G] = ft(!1), Q = st(null);
  Yo(Q, G, X);
  const de = Object.assign({}, {
    /**
     * At what point does the match algorithm give up. A threshold of 0.0
     * requires a perfect match (of both letters and location), a threshold
     * of 1.0 would match anything.
     */
    threshold: 0.05,
    /**
     * Determines approximately where in the text is the pattern expected to be found.
     */
    location: 0,
    /**
     * Determines how close the match must be to the fuzzy location
     * (specified by location). An exact letter match which is distance
     * characters away from the fuzzy location would score as a complete
     * mismatch. A distance of 0 requires the match be at the exact
     * location specified, a distance of 1000 would require a perfect
     * match to be within 800 characters of the location to be found
     * using a threshold of 0.8.
     */
    distance: 100,
    /**
     * When set to include matches, only the matches whose length exceeds this
     * value will be returned. (For instance, if you want to ignore single
     * character index returns, set to 2).
     */
    minMatchCharLength: 1,
    /**
     * List of properties that will be searched. This supports nested properties,
     * weighted search, searching in arrays of strings and objects.
     */
    keys: ["value"]
  }, n), Z = new Ve(r, de), q = (ne) => {
    const { value: te } = ne.target, ve = Z.search(te);
    X(te), k(ve), G(!0), c && c(te);
  }, Y = () => /* @__PURE__ */ ke(
    Bo,
    {
      placeholder: e,
      name: t,
      value: g,
      onChange: q,
      autoFocus: o || !1,
      onFocus: a || void 0,
      inputFontColor: f,
      inputBorderColor: d,
      inputFontSize: h,
      inputHeight: y,
      inputBackgroundColor: l,
      leftIcon: R,
      iconBoxSize: O,
      onKeyDown: (te) => {
        T.length === 1 && te.code === "Enter" && (te.preventDefault(), i(T[0]));
      },
      type: _
    }
  ), H = (ne) => {
    X(A ? "" : ne.item.value), G(!1), i && i(ne), c && c(ne.item.value);
  }, he = () => J ? /* @__PURE__ */ ke(
    Fo,
    {
      matchedRecords: T,
      onClick: H,
      dropdownHoverColor: w,
      dropdownBorderColor: b
    }
  ) : !1;
  return /* @__PURE__ */ ke("div", { ref: Q, children: /* @__PURE__ */ ur(Go, { children: [
    /* @__PURE__ */ ke(Uo, {}),
    Y(),
    he()
  ] }) });
};
export {
  Ko as default
};
